<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">


<meta name="wpd_version" content="0.2">
<meta name="wpd_baseurl" content="http://www.sciencedirect.com/science/article/pii/S0167865510000954">
<meta name="wpd_url" content="http://www.sciencedirect.com/science/article/pii/S0167865510000954">
<meta name="wpd_date" content="2016-01-19T12:25Z">
<script src="geo2.js" async="" type="text/javascript"></script>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">

<meta name="robots" content="NOARCHIVE">
<meta name="robots" content="noodp">
<meta name="robots" content="noydir">

<link rel="canonical" href="http://www.sciencedirect.com/science/article/pii/S0167865510000954">


<link rel="shortcut icon" href="favSD.ico" type="image/x-icon">
<title>

 Variable selection using random forests
 </title> 











<!--[if lt IE 9]>
<link rel="stylesheet" Title="Default Styles" href="http://cdn.els-cdn.com/sd/css/css_gen_ielt9_1602R3.css" type="text/css">
<![endif]-->

<script>

var ab=new Object();
ab.entitled = true;
</script>
<script>(function() {var scriptTag = document.createElement('script'); 
scriptTag.type = 'text/javascript';scriptTag.async = true;
scriptTag.src = '//cdn.optimizely.com/js/204774041.js';
var head = document.getElementsByTagName('head')[0];
head.parentNode.insertBefore(scriptTag, head);
})();function optimizelyTimeout() {
window.optimizely = window.optimizely|| []; 
if(!window.optimizely.data) {
window.optimizely.push("timeout"); } }
setTimeout(optimizelyTimeout, 1000); </script>

<script>

var prs = new Object();prs.rt=function(label,ts){this[label]=(ts||ts==0?ts:new Date().getTime());};
</script>



<!--[if lte IE 7]>
<script src="http://cdn.els-cdn.com/sd/ut/js/json2_1602R3.js.min"></script>
<![endif]-->
<script charset="utf-8" class="lazyload" src="shindig-containerpubsub-2.js"></script><script charset="utf-8" class="lazyload" src="SciverseShindigService,GadgetToolBar,jquery-outside,jquery.ba-bbq.min,SciverseShindigContainer,Sc.js"></script>
<link rel="stylesheet" type="text/css" href="S0167865510000954.css" media="all">
</head>
<body><div id="isIElte8test"><!--[if lte IE 8]><div id="isIElte8"></div><![endif]--></div><div id="lightningjs-usabilla_live" style="display: none;"><div><iframe src="update" id="lightningjs-frame-usabilla_live" frameborder="0"></iframe></div></div><div id="isIE9test"><!--[if  IE 9]><div id="isIE9"></div><![endif]--></div><div id="isIE8test"><!--[if  IE 8]><div id="isIE8"></div><![endif]--></div><div id="isIE7test"><!--[if  IE 7]><div id="isIE7"></div><![endif]--></div><div id="isIElt7test"><!--[if lt IE 7]><div id="isIElt7"></div><![endif]--></div>

<div class="ua_btn" role="region" aria-label="screen reader compatability"><a role="button" rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0167865510000954?np=y">Screen reader users, click here to load entire article</a>This page uses JavaScript to progressively load the article content as a user scrolls.  Screen reader users, click the load entire article button to bypass dynamically loaded article content.</div>



<script>

var refLinkURL = "http://www.sciencedirect.com/science/relLink/ajaxReq?_eid=1-s2.0-S0167865510000954";
var updatedCitedBy = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_updateCitedBy=";
var refResolvePath = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_pii=S0167865510000954&_refCnt=38&_docType=article";

// these variables handle the reference resolution. Please ignore the naming.
// the first variable myMap will hold an associative array of links by bibID
// the second variable myXabsCounts holds the scopus counts.
var myMap = []; // html of links for references.
var myXabsCounts = []; // scopus counts
</script>
<div id="header-area">
  


<!-- ************* For A/B testing FAST and SOLR Proxy ***************** -->

<!-- *************End for A/B testing FAST and SOLR Proxy *****************-->



<div id="ieWarningMsg">
    <span class="infoIcon_ie spErrorIcon">

        Please note that Internet Explorer version 8.x will not be supported as of January 1, 2016. Please refer to <a href="http://blog.sciencedirect.com/posts/have-you-heard-spread-the-word-about-upgrading-your-browser"> this blog post</a> for more information.
    </span>
    <a class="closeBt" title="Close Internet Explorer 7 warning">Close</a>
</div>



<div id="header">
<div class="headerBlk">
    <div class="headerInner" role="navigation" aria-label="ScienceDirect">
        <div class="home">
            <iframe style="display: none;" src="update_001.dat"></iframe><iframe style="display: none;" src="update_002.dat"></iframe><iframe style="display: none;" src="update_003.dat"></iframe><iframe style="display: none;" src="update_004.dat"></iframe><iframe style="display: none;" src="S0167865510000954_6.html"></iframe>
            <!--INSERTS THE TEMPLATE VARIABLES POPULATED FROM DB for SCIVERSE and PRODUCT labels-->
            
                <a href="http://www.sciencedirect.com/" title="ScienceDirect -The world's leading full-text scientific database"> <span class="logoScienceDirectImg">&nbsp;</span></a>
            
        </div>

        
            <ul class="publication">
                <!-- Journals link -->
                
                    
                        <li>
                            <a href="http://www.sciencedirect.com/science/journals">Journals</a>
                        </li>
                     
                

                <!-- Books link -->
                
                    
                        <li class="sep">
                            <a href="http://www.sciencedirect.com/science/bookbshsrw">Books</a>
                        </li>
                     
                
            </ul>
        

    </div>
    
         <!-- if not logged in -->
         
            
            <div id="login" class="clearfix" aria-label="Help and Sign In" role="navigation" style="margin-right: 230px;">
            <ul class="userlinks">
                
                
                
            
                <!--top right widget-->
                <li id="loginPlus">
                    <h2><a id="loginPlusScript" href="javascript:LoginBox.flipLogin('loginBox',null);" title="Sign in to ScienceDirect" class="novisit plus nojs S_C_SignIn" aria-haspopup="true" aria-expanded="false" aria-controls="loginBox" role="button">Sign in</a></h2>
                    <noscript><a href="http://www.sciencedirect.com/science?_ob=HomePageURL&_method=userHomePage&_fl=y&md5=6dada1fb726164470bad9e40bbc4d7fa" title="Open the Sign in Box" class="plus S_C_SignIn" id="loginPlusNoScript" >Sign in</a></noscript>
					
                    <div id="loginBox" role="dialog" aria-labelledby="headingSignIn" class="bannerAvail" style="width: 600px; display: none;"> 
                        
                        <div style="width: 285px; float: right; border-left: 1px solid rgb(155, 155, 155); padding-left: 10px; min-height: 180px;">
                            
                                <div class="openAthens"><a href="https://auth.elsevier.com/ShibAuth/institutionLogin?entityID=https%3A%2F%2Fidp.eduserv.org.uk%2Fopenathens&amp;appReturnURL=http%3A%2F%2Fwww.sciencedirect.com%2Fscience%2Farticle%2Fpii%2FS0167865510000954" title="Log in using OpenAthens">OpenAthens login</a></div>
                            
                            
                            <div class="sampletext">
                                <div style="font-weight: bold; padding-bottom: 7px;"><h3><b>Login via your institution</b></h3></div>
                                
                                
                                    <div class="fedLink"><a href="http://www.sciencedirect.com/science/externallogin" title="Other institution login">Other institution login</a></div>
                                
                            </div>
                            
                            
                            
                                
                            
                            
                            
                        </div>
                        
                        <div class="padding" style="width: 285px;">
                            <form name="sdLogin" method="POST" action="https://www.sciencedirect.com/science/login" id="login-box-form" style="margin: 0px;">
                            <b><h3 id="headingSignIn">Sign in using your ScienceDirect credentials</h3></b>
                            <input name="login" value="login" type="hidden">
                            <div style="margin-top: 20px;">
                            <label for="username"><span style="text-align: right;">Username:</span>
                            <input id="username" name="userid" title="Please enter your username here" type="text"></label>
                            </div>
                            
                            <div style="margin-top: 7px;">
                            <label for="password"><span style="text-align: right;">Password:</span>
                            <input id="password" name="password" maxlength="20" title="Please enter your password here" onkeypress="if(13==event.keyCode)this.form.submit();if (13==event.keyCode)return false;" type="password"></label>
                            </div>
                                   
                                       
                            
                            <div class="remember">
                            <input checked="" name="rememberid" id="rememberMe" value="Y" class="allCheck top2fix" title="Checking this option will log you in automatically next time you visit ScienceDirect" type="checkbox">
                            <label for="rememberMe" title="Checking this option will log you in automatically next time you visit ScienceDirect">Remember me</label>
                            </div>
                            
                            <input name="_ob" value="MiamiLogonURL" type="hidden">
                            <input name="targetURL" value="http://www.sciencedirect.com/science/article/pii/S0167865510000954" type="hidden">
                            <input name="returnURL" value="http://www.sciencedirect.com/science/article/pii/S0167865510000954" type="hidden">
                            <div class="submit">
                            
                            <div style="float: left; padding-left: 0px;"><input class="button" name="arrow" alt="Sign in to ScienceDirect" title="Sign in to ScienceDirect" value="Sign in" type="Submit"></div>
                        
                            
                            <div class="notRegistered"><span class="astPad">| </span><span><a href="https://www.sciencedirect.com/science/register-new-user" title="" target="_top">Not Registered?</a></span></div>
                            
                            
                            <div style="clear: left;"></div>
                            </div>
                            
                            <div class="forgotpass">
                            <a href="http://www.sciencedirect.com/science?_ob=ReminderURL&amp;_method=display&amp;md5=4a97631c25ac81c09ad2f961106e9da2" target="passwd" onclick="var remindWin;remindWin=window.open('','passwd','scrollbars=yes,resizable=yes,directories=no,toolbar=n,menubar=no,status=no,width=650,height=300');" title="Click here for help in recovering your ScienceDirect access">Forgotten username or password?</a>
                            </div>
                            </form>
                        </div>
                    </div>
                </li>
                
                <li class="last"><a class="S_C_helpLink" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=browse_open_doc.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" target="sdhelp" title="Help (Opens new window)">Help</a></li>
                
            </ul>
            </div>
            
            
        <!-- logged in -->
             
    
</div>


    
                        
        <div class="clearfix" id="bannerTop" role="banner" aria-label="Institutional Access">                        
            
                <a href="http://portal.uned.es.ezproxy.uned.es/portal/page?_pageid=93,505432,93_25992941&amp;_dad=portal&amp;_schema=PORTAL" onclick="openNS('http://portal.uned.es.ezproxy.uned.es/portal/page?_pageid=93,505432,93_25992941&amp;_dad=portal&amp;_schema=PORTAL'); return false;">
                    <img src="UNED.jpg" border="0">                    
                </a>
            
        </div>
        
    
           


</div><!--header-->









</div><!--header-area-->
<div class="pgBorder" id="page-area">
    
     
     
    <div data-top="42" id="articleToolbar" class="clearfix quicksearch prd_sci_dir fixedPane">
        <div style="margin-left: 237px;" id="articleNav">
  <ul aria-label="Article options" role="navigation" class="clearfix">
    <!-- return to gateway -->
     






<li>
  <div class="icon_pdf toolbar">
    <div class="largeIcon"></div>
    <a id="pdfLink" rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0167865510000954/pdfft?md5=3d9a92097ce9983bbb82fe06aa8da2e7&amp;pid=1-s2.0-S0167865510000954-main.pdf" querystr="?_origin=article&amp;_zone=toolbar" target="newPdfWin" pdfurl="http://www.sciencedirect.com/science/article/pii/S0167865510000954/pdfft?md5=3d9a92097ce9983bbb82fe06aa8da2e7&amp;pid=1-s2.0-S0167865510000954-main.pdf" suggestedarturl="http://www.sciencedirect.com/science/suggestedArt/citeList/pii/S0167865510000954/eid/1-s2.0-S0167865510000954/serial" class="S_C_pdfLink big pdf ext_sdlink cLink" style="cursor: pointer;" title="Download PDF">Download PDF
    <span id="pdfExplanation" class="offscreen">Opens in a new window.  Article suggestions will be shown in a dialog on return to ScienceDirect.</span>
    </a>
  </div>
</li>
 
 







<li class="exportPopupListitem">
<div class="pulldown">
    <div class="exportDivision" title="Export citations">
    <div class="exportIconText">
    <div class="icon_exportarticlesci_dir initialImage export_icon" tabindex="0"> &nbsp; </div>
    <input id="icon_exportarticle" value="Export" name="export" class="listAction export export_general" tabindex="-1" type="button">
     <span class="downSM_sci_dir exportArrow"></span>
    </div>
    <input class="zone" name="zone" value="exportDropDown" type="hidden">
    </div>
    <div class="menu export_popup hidden" id="export_popup">
    <div style="display: none;" class="warningIcon" title="Warning Icon" alt="Warning Icon"></div>
    <form name="exportCite" method="post" action="http://www.sciencedirect.com/science?_ob=DownloadURL&amp;_method=finish&amp;_eidkey=1-s2.0-S0167865510000954&amp;count=1&amp;_docType=FLA&amp;md5=553329047cabd885ef073abec9fab2c5" target="">
    <input class="zone" name="zone" value="exportDropDown" type="hidden">
            <script language="javascript" type="text/javascript">

<!--

function checkExportFormat(limit) {

    var msg = "";

    for (i=0;i<document.exportCite.elements.length;i++) {
        if (document.exportCite.elements[i].value == "REFWX") {
            if (document.exportCite.elements[i].checked) {
                document.exportCite.setAttribute('target',"RefWorksMain");
                if (checkLimit()) {
                    if (limit) {
                        msg = "It is possible that some users may experience problems when downloading large numbers of citations. If this happens please contact RefWorks support at your regionally hosted site or e-mail support@refworks.com to assist you further.\n\n";
                    }
                    msg += "Exporting to RefWorks may take some time, please wait....";
                    alert(msg);
                }
            }
            else {
                document.exportCite.setAttribute('target',"");
            }
            break;
        }
    }
    return true;
}

function determineCount(numStr) {
    var count=1;

    if(numStr.indexOf("-")) {
        var rangeStr = numStr.split("-");

        if(rangeStr.length == 2) {
            var highNum = parseInt(rangeStr[1]);
            var lowNum = parseInt(rangeStr[0]);

            if(highNum >= lowNum) { count = (highNum - lowNum) + 1; }
        }
    }
    return count;
}

function checkLimit(){

    var selectNum = "";
    var useSelected = false;
    var allNum = 0;
    var count = 0;
    var rdoc = 0;

    for (i=0;i<document.exportCite.elements.length;i++) {
        if (document.exportCite.elements[i].name == "count") {
            allNum = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].name == "NUMBER_LIST") {
            selectNum = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].name == "_rdoc") {
            rdoc = document.exportCite.elements[i].value;
        }
        if (document.exportCite.elements[i].value == "selected") {
            if (document.exportCite.elements[i].checked) {
                useSelected = true;
            }
        }
    }

    if (useSelected) {
        selectNum = selectNum.replace(/ /g,"");
        if(selectNum.indexOf(",")) {
            var commaList = selectNum.split(",");
            for(i=0; i < commaList.length; i++) {
                count += determineCount(commaList[i]);
            }
        }
        else {
            count += determineCount(selectNum);
        }
    }
    else { 
        if (rdoc > 0) {
            count = 1;
        }
        else {
            count = allNum;
        }
    }

    if(count > 50) { return true; }
  
    return false;
}

//-->
</script>

    <div class="exportInfo">
        <div class="chkboxCount"></div>
        <div class="exportHelp">
            <a target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=art_export.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)">Help</a>
        </div>
    </div>

    <div class="expContainer">
      <ul class="Export_Popup" width="100%">
        <li class="subHeading"><h3> Direct export </h3></li>
        <li class="direct_export">
         
            <div>
              <div class="btn_export directExport">
                <div class="icon_Mendeley" title="Save to Mendeley">&nbsp;</div>
                <input id="directExport" value="Save to Mendeley" name="Save to Mendeley" alt="Save to Mendeley" class="mendeleyBT button" type="button">
              </div>
            </div>
        
        
            <div class="btnLine">
              <div class="btn_export refworks">
                <div class="icon_Refworks" title="Save to RefWorks">&nbsp;</div>
                <input id="REFWX" value="Save to RefWorks" name="Save to RefWorks" alt="Save to RefWorks" class="button refworks" type="button">
              </div> 
            </div>
            <div class="clear"> </div>
        
        </li>
        <li class="subHeading big"> <h3>Export file</h3></li>
        <li class="export_radio RIS"><input id="RIS" checked="" name="citation-type" value="RIS" class="artRadio expRadio RIS" type="radio">&nbsp;<label for="RIS">RIS&nbsp;(for EndNote, Reference Manager, ProCite)</label>
        </li>
        <li class="export_radio"><input id="BIBTEX" name="citation-type" value="BIBTEX" class="artRadio expRadio BIBTEX" type="radio">&nbsp;<label for="BIBTEX">BibTeX</label>
        </li>
        <li class="export_radio"><input id="ASCII" name="citation-type" value="ASCII" class="artRadio expRadio ASCII" type="radio">&nbsp;<label for="ASCII">Text</label>
        </li>
        <li class="export_radio hidden"><input id="REFWX" name="citation-type" value="REFWX" class="artRadio expRadio REFWX" type="radio">&nbsp;<label for="REFWX">RefWorks Direct Export</label>
        </li>
        <li class="subHeading big"><h3>Content</h3></li>
        <li class="expCitab export_radio"><input id="cite" class="cite artRadio expRadio" name="format" value="cite" type="radio">&nbsp;<label class="cit" for="cite"> Citation Only </label>
        </li>
        <li class="expCitab export_radio citab"><input id="cite-abs" class="cite-abs artRadio expRadio" name="format" value="cite-abs" checked="" type="radio">&nbsp;<label class="cab" for="cite-abs"> Citation and Abstract </label>
        </li>
        <li class="export_submit">
            <div class="downBtn export_downBtn">
               <input value="Export" name="export" alt="Export" class="button export_button" id="export_button" type="submit">
            </div> 
        </li>
      </ul>

    </div>


        
        
            
         

    </form>
    </div> 
</div>    
</li>
  



</ul>
</div>

<div class="sdSearch" aria-label="Quick search" role="search">
    <fieldset>
        <div class="left">
            <form id="quickSearch" name="qkSrch" method="get" target="_top" action="http://www.sciencedirect.com/science">
            <input name="_ob" value="QuickSearchURL" type="hidden">
            <input name="_method" value="submitForm" type="hidden">
            <input name="_acct" value="C000034358" type="hidden">
            <input name="searchtype" value="a" type="hidden">
            <input name="_origin" value="article" type="hidden">
            <input name="_zone" value="qSearch" type="hidden">
            <input name="md5" value="93c3db4f6f8b4c269845e3a690e78323" type="hidden">
            
            <input id="quickSearch" name="qs_all" placeholder="Search ScienceDirect" value="" size="30" maxlength="450" title="Perform a quick search on ScienceDirect" type="text"><input class="submit" value="" title="Submit quick search" type="submit">
            
            <a class="advanced S_C_advanced" href="http://www.sciencedirect.com/science/search">Advanced search</a>
            </form>
        </div>
    </fieldset>
</div>
<div class="clear"></div>
<div id="erToolbar"></div>


<div style="color: rgb(255, 0, 0); font-weight: bold; border: 0px solid red;">
  
  
  
  
</div>





        <!-- PDFKING/PDFCITE -->
        <div id="pdfModalWindow" class="pdfModalWindow" style="display: none;" onclick="pdfCite.closePopup();"></div>
        <div id="suggestedPdfList" style="display: none;">
         
            <div id="suggestedPdfList" style="display: block;" class="newStyles">
  <div id="modalBoxDisplay" class="nojs" tabindex="0" role="dialog" aria-labelledby="hdSDSuggestions">
    <div class="pdfModalWindowMsgBox" id="pdfModalWindowMsgBox">
	  <div class="modal" style="box-shadow: none; margin-left: 12px; margin-right: 12px;">
        <div id="displayPdfKing" style="display: block;">
           <form method="get" name="Tag" action="http://www.sciencedirect.com/science/suggestedArt/export" style="padding: 0px; margin: 0px;" target="">
            <input name="pdfDownload" value="download" type="hidden">
            
                <input name="PDF_DDM_MAX" value="25" type="hidden">
             
            <input name="_retainOrder" value="Y" type="hidden">
            <input class="zone" name="zone" value="DDM" type="hidden">
            <div class="header" style="height: auto; padding-bottom: 20px; background: rgb(255, 255, 255) none repeat ; 0%;">
            <div class="pdfKingHeading" style="display: block;">Other users also viewed these articles
            <a class="newCloseBtn" id="pdfKingcloseBtn" href="#" title="Close Article Suggestions" alt="Close Article Suggestions"><img src="closeButton.png"></a></div></div>
            <div id="SuggestedArtList" style="min-height: 100px;">
                <div class="toolBar">
                <ul class="group"> 
                    
                    <li class="pdfDownloader" style="display: inline-block; padding: 4px 4px 0px; border: 1px solid rgb(201, 201, 201);">
                        <input value="Download All PDFs" name="pdfDownloadButton" class="listAction iconBg" id="multiPdfIcon" aria-describedby="downloadAllPdfAx" style="width: auto; font-size: 14px; color: rgb(49, 108, 157);" type="submit">
                        <span class="arrow"></span>
                        <span id="downloadAllPdfAx" class="hidden">Opens document download manager</span>
                    </li>
                    
                </ul>
                </div>
                
                <div id="pdfkingContent" style="display: none;">
                    <div class="popcontent"></div> 
                    <div id="relatedArtList" data-entitled="Y"></div>
                </div>
                
            </div>
                
                <div class="newFooter">
                <a class="do-not-show" href="http://www.sciencedirect.com/science/suggestedArt/optOut" style="cursor: pointer; float: none; line-height: normal;">Do not show again</a>
                </div>
          </form>
        </div>
        <div id="pdfKingMessage" style="display: none;">
          <div class="pdfKingHeading" style="display: block;">
            <span class="downloadMsg">Downloading zip file containing 3 articles...</span>
            <a class="newCloseBtn" onclick="pdfCite.closePopup();" href="#" title="Close Article Suggestions" alt="Close Article Suggestions"><img src="closeButton.png"></a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
        
        </div>
        <!-- PDFKING/PDFCITE End -->
    </div>
    <div style="display: block; height: 56px;" id="articleToolbarDummy"></div>
  <div style="clear: both;"></div>
  
  <div style="height: 742px;" id="leftPane" class="column" aria-label="table of contents" role="navigation">
    <div class="fixedPane" style="height: 784px; width: 260px; top: 56px; left: 43.5px;" id="leftPaneInner">
        <div>
        <div id="leftButton">
            <ul>
                <li class="tocbutton">
                    <a class="outline-toggle" title="Open to view article outline" alt="Open to view article outline" href="#" aria-expanded="false">
                        <span class="icon outline_big"></span>
                    </a>
                </li>
            </ul>
        </div>
        <div style="top: 0px;" id="outlineContainer">
            <a class="closeBt" title="Close">&nbsp;</a>
            <div data-st="0" id="outline" class="outline"><div id="olGraphCbBox"><div class="olHead"><h2 id="toc_title">Article outline</h2><label style="display: block;"><input id="outlineGraphicsCheckBox" type="checkBox">Show full outline</label></div><ul id="itemOutline"><li class="li_h2"><div id="ol_author1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#author1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">Abstract<br></a></div></li><li class="li_h2"><div id="ol_kwd_1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#kwd_1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">Keywords<br></a></div></li><li class="li_h2"><div id="ol_section_lbl1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1. Introduction<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.1" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1.1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1.1. Random forests<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.2" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1.2" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1.2. Variable importance<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1.3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1.3. Variable selection<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.4" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1.4" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1.4. A typical situation<br></a></div></li><li class="li_h3"><div id="ol_section_lbl1.5" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl1.5" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">1.5. Outline<br></a></div></li><li class="li_h2"><div id="ol_sec1" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#sec1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">2. Variable importance<br></a></div></li><li class="li_h3"><div id="ol_section_lbl2.1" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl2.1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">2.1. Sensitivity to <em>n</em> and <em>p</em><br></a></div></li><li class="li_h3"><div id="ol_sec2" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#sec2" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">2.2. Sensitivity to <em>mtry</em> and <em>ntree</em><br></a></div></li><li class="li_h3"><div id="ol_section_lbl2.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl2.3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">2.3. Sensitivity to highly correlated predictors<br></a></div></li><li class="li_h3"><div id="ol_section_lbl2.4" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl2.4" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">2.4. Prostate data variable importance<br></a></div></li><li class="li_h2 outlineFocusFirst outlineFocus"><div id="ol_sec3" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#sec3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">3. Variable selection<br></a></div></li><li class="li_h3"><div id="ol_section_lbl3.1" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl3.1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">3.1. Procedure<br></a></div></li><li class="li_h3"><div id="ol_section_lbl3.2" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl3.2" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">3.2. Starting example<br></a></div></li><li class="li_h3"><div id="ol_section_lbl3.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl3.3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">3.3. Highly correlated variables<br></a></div></li><li class="li_h2 outlineFocus outlineFocusLast"><div id="ol_sec4" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#sec4" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">4. Experimental results<br></a></div></li><li class="li_h3"><div id="ol_sec5" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#sec5" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">4.1. Prostate data<br></a></div></li><li class="li_h3"><div id="ol_sec6" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#sec6" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">4.2. Four high dimensional classification datasets<br></a></div></li><li class="li_h3"><div id="ol_section_lbl4.3" class="io item h3sec"><a class="tocLink S_C_tocLinkSec" href="#section_lbl4.3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">4.3. Ozone data<br></a></div></li><li class="li_h2"><div id="ol_sec7" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#sec7" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">5. Discussion<br></a></div></li><li class="li_h2"><div id="ol_bibl001" class="io item h2sec"><a class="tocLink S_C_tocLinkSec" href="#bibl001" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial');">References<br></a></div></li></ul><script>prs.rt('scr_toc')</script></div>
              <script type="text/javascript">

              document.write("<div class='outlineMsg'>Article outline is loading...</div>");
              </script>
              <noscript>

              <div class="infoMsg">
              <div style="float: left; width: 20px; "><img border="0" src="http://cdn.els-cdn.com/sd/alertIcon.gif" valign="top" alt="Alert message"></div>
              <div style="padding-left:25px;">JavaScript is disabled on your browser. 
<div style="padding-top:10px;">
              Please enable JavaScript to use all the features on this page.</div></div>
              </div>
              </noscript>
            <div id="fig" role="navigation" aria-labelledby="fig_title"><hr><h2 id="fig_title">Figures and tables</h2><ol class="figures " role="tree"><li><div id="ol_fig1" class="io item fig" data-parent="sec1"><a class="tocLink S_C_tocLinkFig" href="#fig1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr1.sml" class="displayImg" title="Variable importance sensitivity to n and p (toys data)." alt="Variable importance sensitivity to n and p (toys data)." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr1.sml" data-thumbheight="155"></a></div></li><li><div id="ol_fig2" class="io item fig" data-parent="sec2"><a class="tocLink S_C_tocLinkFig" href="#fig2" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr2.sml" class="displayImg" title="Variable importance sensitivity to mtry and ntree (toys data)." alt="Variable importance sensitivity to mtry and ntree (toys data)." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr2.sml" data-thumbheight="157"></a></div></li><li><div id="ol_fig3" class="io item fig" data-parent="section_lbl2.3"><a class="tocLink S_C_tocLinkFig" href="#fig3" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr3.sml" class="displayImg" title="Variable importance in presence of a group of correlated variables (augmented ..." alt="Variable importance in presence of a group of correlated variables (augmented ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr3.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig4" class="io item fig" data-parent="section_lbl2.3"><a class="tocLink S_C_tocLinkFig" href="#fig4" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr4.sml" class="displayImg" title="Variable importance in presence of two groups of correlated variables (augmented ..." alt="Variable importance in presence of two groups of correlated variables (augmented ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr4.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig5" class="io item fig" data-parent="section_lbl2.4"><a class="tocLink S_C_tocLinkFig" href="#fig5" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr5.sml" class="displayImg" title="Variable importance for Prostate data (using ntree=2000 and mtry=p/3, on the ..." alt="Variable importance for Prostate data (using ntree=2000 and mtry=p/3, on the ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr5.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig6" class="io item fig" data-parent="section_lbl3.2"><a class="tocLink S_C_tocLinkFig" href="#fig6" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr6.sml" class="displayImg" title="Variable selection procedures for interpretation and prediction for toys data." alt="Variable selection procedures for interpretation and prediction for toys data." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr6.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig7" class="io item fig" data-parent="section_lbl3.2"><a class="tocLink S_C_tocLinkFig" href="#fig7" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr7.sml" class="displayImg" title="Selecting the threshold for variable elimination using CART. Bold line refers to ..." alt="Selecting the threshold for variable elimination using CART. Bold line refers to ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr7.sml" data-thumbheight="160"></a></div></li><li><div id="ol_tbl1" class="io item tbl" data-parent="section_lbl3.3"><a class="tocLink olIcon olIconTbl S_C_tocLinkTbl" title="Variable selection procedures in presence of highly correlated variables ..." alt="Variable selection procedures in presence of highly correlated variables ..." href="#tbl1" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial'); "><div class="olIcon olIconTbl"></div>Table 1</a></div></li><li><div id="ol_fig8" class="io item fig" data-parent="sec5"><a class="tocLink S_C_tocLinkFig" href="#fig8" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr8.sml" class="displayImg" title="Variable selection procedures for interpretation and prediction for Prostate ..." alt="Variable selection procedures for interpretation and prediction for Prostate ..." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr8.sml" data-thumbheight="163"></a></div></li><li><div id="ol_tbl2" class="io item tbl" data-parent="sec6"><a class="tocLink olIcon olIconTbl S_C_tocLinkTbl" title="Variable selection procedure for four high dimensional real datasets. CV-error ..." alt="Variable selection procedure for four high dimensional real datasets. CV-error ..." href="#tbl2" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection', 'article', 'leftPane', 'serial'); "><div class="olIcon olIconTbl"></div>Table 2</a></div></li><li><div id="ol_fig9" class="io item fig" data-parent="section_lbl4.3"><a class="tocLink S_C_tocLinkFig" href="#fig9" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr9.sml" class="displayImg" title="Variable importance for ozone data." alt="Variable importance for ozone data." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr9.sml" data-thumbheight="164"></a></div></li><li><div id="ol_fig10" class="io item fig" data-parent="section_lbl4.3"><a class="tocLink S_C_tocLinkFig" href="#fig10" onclick="SD_UTIL.sendUserKeyEvent('displayTOCSection','article','leftPane', 'serial');"><img data-loaded="true" src="1-s2.0-S0167865510000954-gr10.sml" class="displayImg" title="Variable selection procedures for interpretation and prediction for ozone data." alt="Variable selection procedures for interpretation and prediction for ozone data." data-smlsrc="http://ars.els-cdn.com/content/thumbimage/1-s2.0-S0167865510000954-gr10.sml" data-thumbheight="164"></a></div></li></ol></div><div id="articleLeftAd" class="adBlock"><iframe class="osztpmsfnnhahikupuwr" scrolling="no" border="0" cellspacing="0" src="S0167865510000954_7.html" frameborder="0"></iframe></div></div>
        </div>
        </div>
    </div>
  </div>
  <div tabindex="-1" id="centerPane" class="column">
    <div style="top: 56px; left: 303.5px;" class="topBorder fixedPane"></div>
    <div id="centerContent" class="centerContent">
    <div id="centerInner" class="centerInner svBigBox" aria-label="Article" role="main">
        
      
      <noscript>

      <div class="msgBox" style="margin-top:10px;">
          <span class="errMsg">
            <div>JavaScript is disabled on your browser.  Please enable JavaScript to use all the features on this page. This page uses JavaScript to progressively load the article content as a user scrolls. Click the View full text link to bypass dynamically loaded article content. <a rel="nofollow" href="http://www.sciencedirect.com/science/article/pii/S0167865510000954?np=y">View full text</a></div>
         </span>
      </div><br/>
      </noscript>
      <div class="head headTemplate page_fragment_ind">

  
<div class="publisherLogo">
  <span class="spElsevierPubIcon"></span>
</div>


  
    <!--?xml version="1.0" encoding="UTF-8"?--><div xmlns:bk="http://www.elsevier.com/xml/bk/dtd" class="publicationHead"><div class="title"><a title="Go to Pattern Recognition Letters on ScienceDirect" href="http://www.sciencedirect.com/science/journal/01678655" querystr="?zone=centerPane" class="cLink"><span>Pattern Recognition Letters</span></a></div><p class="volIssue"><a class="S_C_volIss" title="Go to table of contents for this volume/issue" href="http://www.sciencedirect.com/science/journal/01678655/31/14">Volume 31, Issue 14</a>, 15 October 2010, Pages 2225–2236</p></div>
  
    <div class="publicationCover cover2">
    
      
    <a querystr="?zone=publicationLogo" class="cLink" href="http://www.sciencedirect.com/science/journal/01678655/31/14">
      
    
    <img alt="Cover image" src="1-s2.0-S0167865510X00125-cov150h.gif" class="toprightlogo">
    <noscript><img alt="Cover image" src="http://ars.els-cdn.com/content/image/1-s2.0-S0167865510X00125-cov150h.gif" class="toprightlogo" /></noscript>
    </a>
    
  </div>


</div>

      <div id="frag_1" class="page_fragment auth_frag" data-first="true" data-fid="1"><div class="module_topic_paths"></div><h1 class="svTitle" id="">Variable selection using random forests</h1><ul style="display: block;" class="authorGroup noCollab svAuthor"><li class="smh5"><a href="#" class="authorName svAuthor" data-orcid="" id="authname_N3ef4d9c0N3efdfbf0" data-t="a" data-fn="Robin" data-ln="Genuer" data-pos="1" data-tb="">Robin Genuer</a><a title="Affiliation: a" href="#aff1" class="intra_ref auth_aff" id="baff1"><sup>a</sup></a><sup>, </sup><a href="mailto:Robin.Genuer@math.u-psud.fr" class="auth_mail" title="E-mail the corresponding author"></a>, </li><li class="smh5"><a href="#" class="authorName svAuthor" data-orcid="" id="authname_N3ef4d9c0N3efdfc80" data-t="a" data-fn="Jean-Michel" data-ln="Poggi" data-pos="2" data-tb="">Jean-Michel Poggi</a><a title="Affiliation: a" href="#aff1" class="intra_ref auth_aff" id="baff1"><sup>a</sup></a><sup>, </sup><a title="Affiliation: b" href="#aff2" class="intra_ref auth_aff" id="baff2"><sup>b</sup></a><sup>, </sup><a title="Corresponding author contact information" href="#cor1" id="bcor1" class="intra_ref auth_corr"></a><sup>, </sup><a href="mailto:Jean-Michel.Poggi@math.u-psud.fr" class="auth_mail" title="E-mail the corresponding author"></a>, </li><li class="smh5"><a href="#" class="authorName svAuthor" data-orcid="" id="authname_N3ef4d9c0N3efdfd10" data-t="a" data-fn="Christine" data-ln="Tuleau-Malot" data-pos="3" data-tb="">Christine Tuleau-Malot</a><a title="Affiliation: c" href="#aff3" class="intra_ref auth_aff" id="baff3"><sup>c</sup></a><sup>, </sup><a href="mailto:malot@unice.fr" class="auth_mail" title="E-mail the corresponding author"></a></li></ul><!--VALIDHTML--><ul style="display: none;" class="affiliation authAffil smh"><li id="aff1"><sup>a</sup> <span id="">Laboratoire de Mathématiques, Université Paris-Sud 11, Bât. 425, 91405 Orsay, France</span></li><li id="aff2"><sup>b</sup> <span id="">Université Paris 5 Descartes, France</span></li><li id="aff3"><sup>c</sup> <span id="">Laboratoire Jean-Alexandre Dieudonné, Université Nice Sophia-Antipolis, Parc Valrose, 06108 Nice Cedex 02, France</span></li></ul><!--VALIDHTML--><!--VALIDHTML--><div id="currtAsOf"></div><dl style="display: none;" class="articleDates smh"><dd>Available online 21 March 2010</dd></dl><!--VALIDHTML--><div id="showMoreButtons"><a class="showLess moreLink"><span class="CollapseText">&nbsp;</span><span class="showInfo expand" title="Show more author and article information">Show more</span></a><a class="showLess lessLink"><span style="display: none;" class="ExpandText">&nbsp;</span><span class="showInfo collapse" title="Hide more author and article information">Show less</span></a></div><!--VALIDHTML--><dl class="extLinks nonEmpty"><dd class="doiLink"><dd class="doi"><a id="ddDoi" class="S_C_ddDoi" href="http://dx.doi.org/10.1016/j.patrec.2010.03.014" target="doilink" onclick="var doiWin; doiWin=window.open('http://dx.doi.org/10.1016/j.patrec.2010.03.014','doilink','scrollbars=yes,resizable=yes,directories=yes,toolbar=yes,menubar=yes,status=yes'); doiWin.focus()">doi:10.1016/j.patrec.2010.03.014</a></dd></dd><dd class="rightsLink"><a class=" cExLink S_C_outLinks" href="https://s100.copyright.com/AppDispatchServlet?publisherName=ELS&amp;contentID=S0167865510000954&amp;orderBeanReset=true" data-url="/science/RedirectURL?_method=outwardLink&amp;_partnerName=936&amp;_eid=1-s2.0-S0167865510000954&amp;_pii=S0167865510000954&amp;_origin=article&amp;_zone=art_page&amp;_targetURL=https://s100.copyright.com/AppDispatchServlet?publisherName%3DELS%26contentID%3DS0167865510000954%26orderBeanReset%3Dtrue" target="_top" title="Get rights and content"><span class="outText">Get rights and content</span></a></dd></dl><div class="moreInformation"><dl class="citation"></dl></div><div class="articleOAlabelForced"></div><div id="refersToAndreferredToBy"><dl id="referredToBy" class="documentThread"><!--Referred To By--></dl></div><!--FRAGMENTEND--><div class="page_fragment_ind auth_frag" data-id="frag_2"></div></div><div id="frag_2" class="page_fragment" data-fid="2"><script>prs.rt("abs_1st_beg");</script><!--VALIDHTML--><hr id="abs_author1" class="artHeader"><div class="abstract svAbstract " data-etype="ab"><h2 class="secHeading" id="author1">Abstract</h2><p id="">This paper proposes, focusing on random forests, the increasingly used statistical method for classification and regression problems introduced by Leo Breiman in 2001, to investigate two classical issues of variable selection. The first one is to find important variables for interpretation and the second one is more restrictive and try to design a good parsimonious prediction model. The main contribution is twofold: to provide some experimental insights about the behavior of the variable importance index based on random forests and to propose a strategy involving a ranking of explanatory variables using the random forests score of importance and a stepwise ascending variable introduction strategy.</p></div><!--VALIDHTML--><h2 id="kwd_1" class="svKeywords">Keywords</h2><ul class="keyword" id=""><li id="" class="svKeywords"><span id="">Random forests</span>; </li><li id="" class="svKeywords"><span id="">Regression</span>; </li><li id="" class="svKeywords"><span id="">Classification</span>; </li><li id="" class="svKeywords"><span id="">Variable importance</span>; </li><li id="" class="svKeywords"><span id="">Variable selection</span>; </li><li id="" class="svKeywords"><span id="">High dimensional data</span></li></ul><!--VALIDHTML--><div id="embeddedPDF" style="display: none;"></div><script>prs.rt("abs_end");</script><div id="SD_BA1P" class="sgfNoTitleBar sgfNoGadgetBorder svDoNotLink"></div><!--VALIDHTML--><hr class="artHeader"><h2 class="svArticle" id="section_lbl1">1. Introduction</h2><p class="svArticle section clear" id="">This paper is primarily interested in random forests for variable selection. Mainly methodological the main contribution is twofold: to provide some experimental insights about the behavior of the variable importance index based on random forests and to use it to propose a two-steps algorithm for two classical problems of variable selection starting from variable importance ranking. The first problem is to find important variables for interpretation and the second one is more restrictive and try to design a good parsimonious prediction model. The general strategy involves a ranking of explanatory variables using the random forests score of importance and a stepwise ascending variable introduction strategy. Let us mention that we propose an heuristic strategy which does not depend on specific model hypotheses but based on data-driven thresholds to take decisions.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Before entering into details, we introduce the three main topics of this paper (random forests, variable importance, variable selection) and we sketch a typical high dimensional classification problem motivating this work.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.1">1.1. Random forests</h3><p class="svArticle section clear" id="">Random forests (RF henceforth) is a popular and very efficient algorithm, based on model aggregation ideas, for both classification and regression problems, introduced by <span id="bbib7"><a href="#bib7" id="ancbbib7" class="intra_ref">Breiman (2001)</a></span>. It belongs to the family of ensemble methods, appearing in machine learning at the end of nineties (see for example <a href="#bib15" id="bbib15" class="intra_ref">Dietterich, 2000a</a>&nbsp;and&nbsp;<a href="#bib16" id="bbib16" class="intra_ref">Dietterich, 2000b</a>). Let us briefly recall the statistical framework by considering a learning set <span id="mmlsi16" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si16.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=282608be434566c5ebc616d81ab476ba" title="Click to view the MathML source">L={(X<sub>1</sub>,Y<sub>1</sub>),…,(X<sub>n</sub>,Y<sub>n</sub>)}</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si16.gif" overflow="scroll"><mrow><mi>L</mi><mo>=</mo><mo stretchy="false">{</mo><mo stretchy="false">(</mo><msub><mrow><mi>X</mi></mrow><mrow><mn>1</mn></mrow></msub><mtext>,</mtext><msub><mrow><mi>Y</mi></mrow><mrow><mn>1</mn></mrow></msub><mo stretchy="false">)</mo><mtext>,</mtext><mo>…</mo><mtext>,</mtext><mo stretchy="false">(</mo><msub><mrow><mi>X</mi></mrow><mrow><mi>n</mi></mrow></msub><mtext>,</mtext><msub><mrow><mi>Y</mi></mrow><mrow><mi>n</mi></mrow></msub><mo stretchy="false">)</mo><mo stretchy="false">}</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> made of <em>n &nbsp;</em> i.i.d. observations of a random vector <span id="mmlsi17" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si17.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=fba3f82377a9154b0446e4fc1cd34d12" title="Click to view the MathML source">(X,Y)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si17.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>X</mi><mtext>,</mtext><mi>Y</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. Vector <span id="mmlsi18" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si18.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=bb9d5d34f9872d12b4ef9941cf60dc7a" title="Click to view the MathML source">X=(X<sup>1</sup>,…,X<sup>p</sup>)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si18.gif" overflow="scroll"><mrow><mi>X</mi><mo>=</mo><mo stretchy="false">(</mo><msup><mrow><mi>X</mi></mrow><mrow><mn>1</mn></mrow></msup><mtext>,</mtext><mo>…</mo><mtext>,</mtext><msup><mrow><mi>X</mi></mrow><mrow><mi>p</mi></mrow></msup><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> contains predictors or explanatory variables, say <span id="mmlsi19" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si19.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ea386db64958e46321d032fe9e624b6a" title="Click to view the MathML source">X∈R<sup>p</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si19.gif" overflow="scroll"><mrow><mi>X</mi><mo>∈</mo><msup><mrow><mi mathvariant="double-struck">R</mi></mrow><mrow><mi>p</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, and <span id="mmlsi20" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si20.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=a128b4112d2b4e496eaabaee14772fb8" title="Click to view the MathML source">Y∈Y</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si20.gif" overflow="scroll"><mrow><mi>Y</mi><mo>∈</mo><mi mathvariant="script">Y</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> where <span id="mmlsi21" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si21.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=0538c0249107455f31ffaccfa08fd8b6" title="Click to view the MathML source">Y</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si21.gif" overflow="scroll"><mrow><mi mathvariant="script">Y</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is either a class label or a numerical response. For classification problems, a classifier <em>t &nbsp;</em> is a mapping <span id="mmlsi22" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si22.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=8e6fcc28bc446d77ff5305545a785bb5" title="Click to view the MathML source">t:R<sup>p</sup>→Y</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si22.gif" overflow="scroll"><mrow><mi>t</mi><mo>:</mo><msup><mrow><mi mathvariant="double-struck">R</mi></mrow><mrow><mi>p</mi></mrow></msup><mo>→</mo><mi mathvariant="script">Y</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> while for regression problems, we suppose that <span id="mmlsi23" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si23.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ff4423c49a8c062550f070c01ab00a16" title="Click to view the MathML source">Y=s(X)+ε</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si23.gif" overflow="scroll"><mrow><mi>Y</mi><mo>=</mo><mi>s</mi><mo stretchy="false">(</mo><mi>X</mi><mo stretchy="false">)</mo><mo>+</mo><mi>ε</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> with <span id="mmlsi24" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si24.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=70f71a5b16ca0a065aa7f374be96d3fd" title="Click to view the MathML source">E[ε|X]=0</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si24.gif" overflow="scroll"><mrow><mi>E</mi><mo stretchy="false">[</mo><mi>ε</mi><mo stretchy="false">|</mo><mi>X</mi><mo stretchy="false">]</mo><mo>=</mo><mn>0</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <em>s</em> the so-called regression function (for more background on statistical learning, see e.g.  <span id="bbib27"><a href="#bib27" id="ancbbib27" class="intra_ref">Hastie et al. (2001)</a></span>). Random forests is a model building strategy providing estimators of either the Bayes classifier, which is the mapping minimizing the classification error <span id="mmlsi25" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si25.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=af868cf35c78c4d1e73e56c7462f3b43"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si25.gif" data-inlimgeid="1-s2.0-S0167865510000954-si25.gif" height="17" width="86"><noscript><img height="17" border="0" style="vertical-align:bottom" width="86" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si25.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si25.gif" overflow="scroll"><mrow><mi>P</mi><mo stretchy="false">(</mo><mi>Y</mi><mspace width="0.25em"></mspace><mo>≠</mo><mspace width="0.25em"></mspace><mi>t</mi><mo stretchy="false">(</mo><mi>X</mi><mo stretchy="false">)</mo><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, or the regression function.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The principle of random forests is to combine many binary decision trees built using several bootstrap samples coming from the learning sample <em>L</em> and choosing randomly at each node a subset of explanatory variables <em>X</em>. More precisely, with respect to the well-known CART<span id="bfn1"><a href="#fn1" id="ancbfn1" class="intra_ref"><sup>1</sup></a></span> model building strategy (see <span id="bbib10"><a href="#bib10" id="ancbbib10" class="intra_ref">Breiman et al. (1984)</a></span>) performing a growing step followed by a pruning one, two differences can be noted. First, at each node, a given number (denoted by <em>mtry</em>) of input variables are randomly chosen and the best split is calculated only within this subset. Second, no pruning step is performed so all the trees of the forest are maximal trees.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_3"></div><script>prs.rt("abs_1st_end");</script></div><div id="frag_3" class="page_fragment" data-fid="3"><p class="svArticle section clear" id="">In addition to CART, bagging, another well-known related tree-based method, is to be mentioned (see <span id="bbib6"><a href="#bib6" id="ancbbib6" class="intra_ref">Breiman (1996)</a></span>). Indeed random forests with <span id="mmlsi26" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si26.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=76974a5a3b4f22e9f09b052592d7dc74" title="Click to view the MathML source"><em>mtry</em>=p</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si26.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> reduce simply to unpruned bagging. The associated R packages<span id="bfn2"><a href="#fn2" id="ancbfn2" class="intra_ref"><sup>2</sup></a></span> are respectively <tt>randomForest</tt> (intensively used in the sequel of the paper), <tt>rpart</tt> and <tt>ipred</tt> for CART and bagging respectively (cited here for the sake of completeness).</p><!--VALIDHTML--><p class="svArticle section clear" id="">RF algorithm becomes more and more popular and appears to be very powerful in a lot of different applications (see for example <span id="bbib14"><a href="#bib14" id="ancbbib14" class="intra_ref">Díaz-Uriarte and Alvarez de Andrés (2006)</a></span> for gene expression data analysis) even if it is not clearly elucidated from a mathematical point of view (see the recent paper by <span id="bbib5"><a href="#bib5" id="ancbbib5" class="intra_ref">Biau et al. (2008)</a></span> about purely random forests and <span id="bbib11"><a href="#bib11" id="ancbbib11" class="intra_ref">Bühlmann and Yu (2002)</a></span> about bagging). Nevertheless, <span id="bbib7"><a href="#bib7" id="ancbbib7" class="intra_ref">Breiman (2001)</a></span> sketches an explanation of the good performance of random forests related to the good quality of each tree (at least from the bias point of view) together with the small correlation among the trees of the forest, where the correlation between trees is defined as the ordinary correlation of predictions on so-called out-of-bag (OOB henceforth) samples. The OOB sample is the set of observations which are not used for building the current tree. It is used to estimate the prediction error and then to evaluate variable importance.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The R package about random forests is based on the seminal contribution of <span id="bbib9"><a href="#bib9" id="ancbbib9" class="intra_ref">Breiman and Cutler (2005)</a></span> and is described in <span id="bbib30"><a href="#bib30" id="ancbbib30" class="intra_ref">Liaw and Wiener (2002)</a></span>. In this paper, we focus on the <tt>randomForest</tt> procedure. The two main parameters are <em>mtry</em>, the number of input variables randomly chosen at each split and <em>ntree</em>, the number of trees in the forest. Some details about numerical and sensitivity experiments can be found in  <span id="bbib20"><a href="#bib20" id="ancbbib20" class="intra_ref">Genuer et al. (2008)</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In addition, we will concentrate on the prediction performance of RF focusing on out-of-bag (OOB) error (see <span id="bbib7"><a href="#bib7" id="ancbbib7" class="intra_ref">Breiman (2001)</a></span>). We use this kind of prediction error estimate for three reasons: the main is that we are mainly interested in comparing models instead of assessing models, the second is that it gives fair estimation compared to the usual alternative test set error even if it is considered as a little bit optimistic and the last one, is that it is a default output of the <tt>randomForest</tt> procedure, so it is used by almost all users.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.2">1.2. Variable importance</h3><p class="svArticle section clear" id="">The quantification of the variable importance (VI henceforth) is an important issue in many applied problems complementing variable selection by interpretation issues.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In the random forests framework, the most widely used score of importance of a given variable is the increasing in mean of the error of a tree (mean square error (MSE) for regression and misclassification rate for classification) in the forest when the observed values of this variable are randomly permuted in the OOB samples (it could be slightly negative). Often, such random forests VI is called permutation importance indices in opposition to total decrease of node impurity measures already introduced in the seminal book about CART by <span id="bbib10"><a href="#bib10" id="ancbbib10" class="intra_ref">Breiman et al. (1984)</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">For regression problems, two other measures of VI are to be mentioned. In the linear regression framework it is examined for example by <span id="bbib23"><a href="#bib23" id="ancbbib23" class="intra_ref">Grömping (2007)</a></span>, making a distinction between various variance decomposition based indicators: “dispersion importance”, “level importance” or “theoretical importance” quantifying explained variance or changes in the response for a given change of each regressor. An extension to nonlinear regression models focusing on the input–output analysis (avoiding model estimation) is provided by Sobol sensitivity indices (<span id="bbib35"><a href="#bib35" id="ancbbib35" class="intra_ref">Sobol’ (1993)</a></span>).</p><!--VALIDHTML--><p class="svArticle section clear" id="">A comparison between RF variable importance and linear regression based importance indices has been carried out by <span id="bbib24"><a href="#bib24" id="ancbbib24" class="intra_ref">Grömping (2009)</a></span> and the conclusion is that the results are in agreement. A preliminary comparison with sensitivity indices (not reported here) lead to the same conclusion. A more intensive comparison could be of interest but it is out of the scope of this paper which focuses on RF. We emphasize that, due to the versatility of the RF framework, RF variable importance can be computed for standard <span id="mmlsi27" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si27.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=05dcbeeccf5e5d7b7897828b212408f4" title="Click to view the MathML source">(n&gt;&gt;p)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si27.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>&gt;</mo><mo>&gt;</mo><mi>p</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> or high dimensional <span id="mmlsi28" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si28.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d92f2053e934d1c2bb7a2f01a05c160c" title="Click to view the MathML source">(n&lt;&lt;p)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si28.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>&lt;</mo><mo>&lt;</mo><mi>p</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> problems, as well as for classification or regression problems. In addition, the computational burden is acceptable.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Even if only little investigation is available about RF variable importance, some interesting facts are collected for classification problems when this index is based on the average loss of heterogeneity criterion, derived for example from the Gini impurity function used for growing classification trees. First, the RF Gini importance is not fair in favor of predictor variables with many categories (see <span id="bbib36"><a href="#bib36" id="ancbbib36" class="intra_ref">Strobl et al. (2007)</a></span>) while the RF permutation importance is a more reliable indicator. So we restrict our attention to this last one. Then, it seems that permutation importance overestimates the variable importance of highly correlated variables and a conditional variant is proposed by <span id="bbib37"><a href="#bib37" id="ancbbib37" class="intra_ref">Strobl et al. (2008)</a></span>. In this paper, we do not diagnose such a critical phenomenon for variable selection. Finally, the recent paper by <span id="bbib3"><a href="#bib3" id="ancbbib3" class="intra_ref">Archer and Kimes (2008)</a></span>, focusing more specifically on the VI topic is also of interest. We give some experimental insights about variable importance behavior in presence of groups of highly correlated variables. This is the first goal of this paper.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.3">1.3. Variable selection</h3><p class="svArticle section clear" id="">Many variable selection procedures are based on the cooperation of variable importance for ranking and model estimation to generate, evaluate and compare a family of models. Following <a href="#bib29" id="bbib29" class="intra_ref">Kohavi and John, 1997</a>&nbsp;and&nbsp;<a href="#bib25" id="bbib25" class="intra_ref">Guyon and Elisseff, 2003</a>, it is usual to distinguish three types of variable selection methods: “filter” for which the score of variable importance does not depend on a given model design method; “wrapper” which include the prediction performance in the score calculation; and finally “embedded” which combine more closely variable selection and model estimation.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Let us briefly mention some of them, in the classification case, which are potentially competing tools: of course the wrapper methods based on VI coming from CART, and from random forests. Then some examples of embedded methods: <span id="bbib32"><a href="#bib32" id="ancbbib32" class="intra_ref">Poggi and Tuleau (2006)</a></span> propose a method based on CART scores and using stepwise ascending procedure with elimination step; <a href="#bib26" id="bbib26" class="intra_ref">Guyon et al., 2002</a>&nbsp;and&nbsp;<a href="#bib33" id="bbib33" class="intra_ref">Rakotomamonjy, 2003</a>, propose methods based on Support Vector Machines (SVM) scores and using descending elimination. More recently, <span id="bbib4"><a href="#bib4" id="ancbbib4" class="intra_ref">Ben Ishak and Ghattas (2008)</a></span> propose a stepwise variant while <span id="bbib31"><a href="#bib31" id="ancbbib31" class="intra_ref">Park and Hastie (2007)</a></span> propose a LARS<span id="bfn3"><a href="#fn3" id="ancbfn3" class="intra_ref"><sup>3</sup></a></span> type strategy (see <span id="bbib17"><a href="#bib17" id="ancbbib17" class="intra_ref">Efron et al. (2004)</a></span>) for classification problems. Finally we mention a mixed approach, see <span id="bbib19"><a href="#bib19" id="ancbbib19" class="intra_ref">Fan and Lv (2008)</a></span> in regression, ascending in order to avoid to select redundant variables or, for the case <span id="mmlsi29" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si29.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=fb1b16254ad8904d2fe98b0fa9144ce4" title="Click to view the MathML source">n&lt;&lt;p</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si29.gif" overflow="scroll"><mrow><mi>n</mi><mo>&lt;</mo><mo>&lt;</mo><mi>p</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, descending first using a screening procedure to reach a classical situation <span id="mmlsi30" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si30.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ea61bcad4816ac20218c5e6646bece61" title="Click to view the MathML source">n∼p</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si30.gif" overflow="scroll"><mrow><mi>n</mi><mi>∼</mi><mi>p</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, and then ascending using LASSO<span id="bfn4"><a href="#fn4" id="ancbfn4" class="intra_ref"><sup>4</sup></a></span> or SCAD<span id="bfn5"><a href="#fn5" id="ancbfn5" class="intra_ref"><sup>5</sup></a></span>, see <span id="bbib18"><a href="#bib18" id="ancbbib18" class="intra_ref">Fan and Li (2001)</a></span>. We propose in this paper, a two-steps procedure, the second one depends on the objective (interpretation or prediction) while the first one is common. The key point is that it is entirely based on random forests, so fully non parametric and then free from the usual linear framework.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.4">1.4. A typical situation</h3><p class="svArticle section clear" id="">Let us close this section by introducing a typical situation which can be useful to capture the main ideas of this paper. We consider a high dimensional <span id="mmlsi31" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si31.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=fe60509b560e6f3adfc2b5261ab55d5f" title="Click to view the MathML source">(n&lt;&lt;p)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si31.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>&lt;</mo><mo>&lt;</mo><mi>p</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> classification problem for which the predictor variables are associated to a pixel in an image or a 3D location in the brain like in fMRI brain activity classification problems. In such situations, of course it is clear that there is a lot of useless variables and that there exists unknown groups of highly correlated predictors corresponding to brain regions. We emphasize that two distinct objectives about variable selection can be identified: (1) to find important variables highly related to the response variable for interpretation purpose; (2) to find a small number of variables sufficient for a good prediction of the response variable. Key tools combine variable importance thresholding, variable ranking and stepwise introduction of variables. Turning back to our typical situation, an example of the first kind of problem is the determination of entire regions in the brain or a full parcel in an image while an instance of the second one is to exhibit a parsimonious subset of the most discriminant variables within the previously highlighted groups.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl1.5">1.5. Outline</h3><p class="svArticle section clear" id="">The paper is organized as follows. After this introduction, Section <span id="bsec1"><a href="#sec1" id="ancbsec1" class="intra_ref">2</a></span> illustrates RF variable importance behavior, especially in presence of groups of highly correlated explanatory variables. Section <span id="bsec3"><a href="#sec3" id="ancbsec3" class="intra_ref">3</a></span> proposes an ascending procedure for two classical variable selection problems starting from an initial ranking based on the random forests score of importance. Section <span id="bsec4"><a href="#sec4" id="ancbsec4" class="intra_ref">4</a></span> examines some experimental results, by focusing mainly on high dimensional classification datasets and, in order to illustrate the general value of the strategy, it is applied to a standard <span id="mmlsi32" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si32.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=94180292f24162744166ba3b63169a7c" title="Click to view the MathML source">(n&gt;&gt;p)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si32.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>&gt;</mo><mo>&gt;</mo><mi>p</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> regression dataset. Finally Section <span id="bsec7"><a href="#sec7" id="ancbsec7" class="intra_ref">5</a></span> opens discussion about future work.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_4"></div></div><div id="frag_4" class="page_fragment" data-fid="4"><h2 class="svArticle" id="sec1">2. Variable importance</h2><p class="svArticle section clear" id="">The quantification of the variable importance is a crucial issue not only for ranking the variables before a stepwise estimation model but also to interpret data and understand underlying phenomenons in many applied problems.</p><!--VALIDHTML--><p class="svArticle section clear" id="">RF variable importance of <span id="mmlsi33" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si33.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=85681f90b1bf2d197b163110fd5e911f" title="Click to view the MathML source">X<sup>j</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si33.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is defined as follows. For each tree <em>t &nbsp;</em> of the forest, consider the associated <span id="mmlsi34" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si34.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=b56fc86949c315bdc4f7b3a6cd5cb376" title="Click to view the MathML source"><em>OOB</em><sub>t</sub></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si34.gif" overflow="scroll"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> sample (data not included in the boostrap sample used to construct <em>t &nbsp;</em>). Denote by <span id="mmlsi35" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si35.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=1080ec025e6efd5e65ad776c7c6bc2e4" title="Click to view the MathML source"><em>errOOB</em><sub>t</sub></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si35.gif" overflow="scroll"><mrow><msub><mrow><mi mathvariant="italic">errOOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> the error (MSE for regression and misclassification rate for classification) of a single tree <em>t &nbsp;</em> on this <span id="mmlsi36" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si36.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ff848db970bae22e0e63a53d2fa0f39e" title="Click to view the MathML source"><em>OOB</em><sub>t</sub></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si36.gif" overflow="scroll"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> sample. Now, randomly permute the values of <span id="mmlsi37" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si37.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e46f5b772481ef6deb1f27f265a36c63" title="Click to view the MathML source">X<sup>j</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si37.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> in <span id="mmlsi38" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si38.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=fad7b6253ff74430ffb9add6ed54e919" title="Click to view the MathML source"><em>OOB</em><sub>t</sub></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si38.gif" overflow="scroll"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> to get a perturbed sample denoted by <span id="mmlsi39" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si39.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=4131628c130cda7df5904ce09615fe4d"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si39.gif" data-inlimgeid="1-s2.0-S0167865510000954-si39.gif" height="19" width="44"><noscript><img height="19" border="0" style="vertical-align:bottom" width="44" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si39.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si39.gif" overflow="scroll"><mrow><msup><mrow><mover accent="true"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow><mrow><mo stretchy="true">∼</mo></mrow></mover></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and compute <span id="mmlsi40" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si40.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=088a1597ecfe395d7be048e73f847ed1"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si40.gif" data-inlimgeid="1-s2.0-S0167865510000954-si40.gif" height="20" width="66"><noscript><img height="20" border="0" style="vertical-align:bottom" width="66" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si40.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si40.gif" overflow="scroll"><mrow><mi mathvariant="italic">err</mi><msup><mrow><mover accent="true"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow><mrow><mo stretchy="true">∼</mo></mrow></mover></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, the error of predictor <em>t &nbsp;</em> on the perturbed sample. Variable importance of <span id="mmlsi41" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si41.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=b02ba1ccb5e6fae0ebc5dd75ac7d4fc4" title="Click to view the MathML source">X<sup>j</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si41.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is then equal to:</p><div class="formula" id=""><div class="mathml"><span id="mmlsi42" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si42.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=39d2f7309e263a830d8d55f8a420f953"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si42.gif" data-inlimgeid="1-s2.0-S0167865510000954-si42.gif" height="42" width="296"><noscript><img height="42" border="0" style="vertical-align:bottom" width="296" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si42.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si42.gif" overflow="scroll"><mi mathvariant="italic">VI</mi><mo stretchy="false">(</mo><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi></mrow></msup><mo stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mn>1</mn></mrow><mrow><mi mathvariant="italic">ntree</mi></mrow></mfrac><munder><mrow><mo>∑</mo></mrow><mrow><mi>t</mi></mrow></munder><mo stretchy="false">(</mo><mi mathvariant="italic">err</mi><msup><mrow><mover accent="true"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow><mrow><mo stretchy="true">∼</mo></mrow></mover></mrow><mrow><mi>j</mi></mrow></msup><mo>-</mo><msub><mrow><mi mathvariant="italic">errOOB</mi></mrow><mrow><mi>t</mi></mrow></msub><mo stretchy="false">)</mo><mtext>,</mtext></math></span><!--mathContainer--></span><div class="btContainer" style="height: 25px;"><div role="region" aria-label="MathJAX Management" class="mathjax"><a href="javascript:void(0);" role="button" class="mathjax firstFormula" title="Turn MathJax on">Turn<span class="mathjax ">&nbsp;</span><span class="offscreen">MathJax </span>on</a></div></div><!--Loading Mathjax--></span><img class="temp" src="blank.gif"><!--Comment--></div></div>where the sum is over all trees <em>t</em> of the RF and <em>ntree</em> denotes the number of trees of the RF.<p></p><!--VALIDHTML--><p class="svArticle section clear" id="">In this section, we examine the RF variable importance behavior according to three different issues. The first one deals with the sensitivity to the sample size <em>n</em> and the number of variables <em>p</em>. The second examines the sensitivity to method parameters <em>mtry</em> and <em>ntree</em>. This is of interest since a good choice of parameters of RF can help to better discriminate between important and useless variables. In addition, it can increase the stability of VI scores. The third one deals with the variable importance in presence of groups of highly correlated variables.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_5"></div></div><div id="frag_5" class="page_fragment" data-fid="5"><p class="svArticle section clear" id="">To illustrate this discussion, we examine a simulated dataset for the case <span id="mmlsi43" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si43.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ae53cca014a2b1372dd74464c21ebf8d" title="Click to view the MathML source">n&lt;&lt;p</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si43.gif" overflow="scroll"><mrow><mi>n</mi><mo>&lt;</mo><mo>&lt;</mo><mi>p</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, introduced by <span id="bbib38"><a href="#bib38" id="ancbbib38" class="intra_ref">Weston et al. (2003)</a></span> and called “toys data” in the sequel. It is an equiprobable two class problem, <span id="mmlsi44" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si44.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d77e5482e1dee2fbdbd03ef25bd55efd" title="Click to view the MathML source">Y∈{-1,1}</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si44.gif" overflow="scroll"><mrow><mi>Y</mi><mo>∈</mo><mo stretchy="false">{</mo><mo>-</mo><mn>1</mn><mtext>,</mtext><mn>1</mn><mo stretchy="false">}</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, with six true variables, the others being some noise. This example is interesting since it constructs two independent groups of three significant variables (highly, moderately and weakly correlated with response <em>Y</em>) and an additional group of noise variables, uncorrelated with <em>Y</em>. A forward reference to the plots on the left side of  <span id="bfig1"><a href="#fig1" id="ancbfig1" class="intra_ref">Fig. 1</a></span> allows to see the variable importance picture: importances of variables 1–3 are higher than the ones of variables 4–6, i.e. <span id="mmlsi45" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si45.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=1d738959248d1ebe7dda6818e813265a" title="Click to view the MathML source"><em>VI</em>(X<sup>j</sup>)&gt;<em>VI</em>(X<sup>j+3</sup>)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si45.gif" overflow="scroll"><mrow><mi mathvariant="italic">VI</mi><mo stretchy="false">(</mo><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi></mrow></msup><mo stretchy="false">)</mo><mo>&gt;</mo><mi mathvariant="italic">VI</mi><mo stretchy="false">(</mo><msup><mrow><mi>X</mi></mrow><mrow><mi>j</mi><mo>+</mo><mn>3</mn></mrow></msup><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi46" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si46.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3bfb09901cab4b39facf6359621cd1e5" title="Click to view the MathML source">j=1,2,3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si46.gif" overflow="scroll"><mrow><mi>j</mi><mo>=</mo><mn>1</mn><mtext>,</mtext><mn>2</mn><mtext>,</mtext><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. The simulation model is defined through the conditional distribution of the <span id="mmlsi47" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si47.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e21c54ddc1781e39142218631db80f6c" title="Click to view the MathML source">X<sup>i</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si47.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi48" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si48.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=18b436078288b7751a53478a42b85942" title="Click to view the MathML source">Y=y</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si48.gif" overflow="scroll"><mrow><mi>Y</mi><mo>=</mo><mi>y</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>:</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">For the six first variables: with probability 0.7, <span id="mmlsi49" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si49.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=b1e03726cbe0283c552ff3eb87993d11" title="Click to view the MathML source">X<sup>i</sup>∼N(<em>yi</em>,1)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si49.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup><mi>∼</mi><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mi mathvariant="italic">yi</mi><mtext>,</mtext><mn>1</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi50" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si50.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d61362ac055f7c8fa7c676378a57610e" title="Click to view the MathML source">i=1,2,3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si50.gif" overflow="scroll"><mrow><mi>i</mi><mo>=</mo><mn>1</mn><mtext>,</mtext><mn>2</mn><mtext>,</mtext><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi51" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si51.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=30251b57858366f12cdd34c13529f4c2" title="Click to view the MathML source">X<sup>i</sup>∼N(0,1)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si51.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup><mi>∼</mi><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mtext>,</mtext><mn>1</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi52" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si52.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=5fa59eaf3fe0c3138a1a37a9c160b124" title="Click to view the MathML source">i=4,5,6</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si52.gif" overflow="scroll"><mrow><mi>i</mi><mo>=</mo><mn>4</mn><mtext>,</mtext><mn>5</mn><mtext>,</mtext><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>; with probability 0.3, <span id="mmlsi53" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si53.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=48d58c23651fc0a02746dcb0b8fdac47" title="Click to view the MathML source">X<sup>i</sup>∼N(0,1)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si53.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup><mi>∼</mi><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mtext>,</mtext><mn>1</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi54" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si54.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=297d41af4996fd24613175f25b801cd6" title="Click to view the MathML source">i=1,2,3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si54.gif" overflow="scroll"><mrow><mi>i</mi><mo>=</mo><mn>1</mn><mtext>,</mtext><mn>2</mn><mtext>,</mtext><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi55" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si55.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=7a2c0d91ca59976805355e1ee9947395" title="Click to view the MathML source">X<sup>i</sup>∼N(y(i-3),1)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si55.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup><mi>∼</mi><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mi>y</mi><mo stretchy="false">(</mo><mi>i</mi><mo>-</mo><mn>3</mn><mo stretchy="false">)</mo><mtext>,</mtext><mn>1</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi56" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si56.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=a96b4b8aa9dddc62405ee43080835a8b" title="Click to view the MathML source">i=4,5,6</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si56.gif" overflow="scroll"><mrow><mi>i</mi><mo>=</mo><mn>4</mn><mtext>,</mtext><mn>5</mn><mtext>,</mtext><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>.</p></dd><dt class="label">•</dt><dd><p id="">Remaining variables are noise: <span id="mmlsi57" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si57.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=8ab3cc9b94f8e2eca2f69932507723ef" title="Click to view the MathML source">X<sup>i</sup>∼N(0,1)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si57.gif" overflow="scroll"><mrow><msup><mrow><mi>X</mi></mrow><mrow><mi>i</mi></mrow></msup><mi>∼</mi><mi mathvariant="script">N</mi><mo stretchy="false">(</mo><mn>0</mn><mtext>,</mtext><mn>1</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for <span id="mmlsi58" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si58.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=24251b36ede5801c7367e58f72445471" title="Click to view the MathML source">i=7,…,p</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si58.gif" overflow="scroll"><mrow><mi>i</mi><mo>=</mo><mn>7</mn><mtext>,</mtext><mo>…</mo><mtext>,</mtext><mi>p</mi></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>.</p></dd></dl><p></p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig1"><div><dl class="figure" id="fig1" data-t="f"><dt class="autoScroll" data-style="height:385px;width:543px;"><a class="figureLink" href="#gr1" title="Variable importance sensitivity to n and p (toys data)."><img data-loaded="true" style="display: inline; height: 385px; width: 543px;" class="imgLazyJSB figure large nrmImg" alt="Variable importance sensitivity to n and p (toys data)." src="1-s2.0-S0167865510000954-gr1.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr1.sml" data-fulleid="1-s2.0-S0167865510000954-gr1.jpg" data-imgeids="1-s2.0-S0167865510000954-gr1.jpg" data-thumbheight="155" data-thumbwidth="219" data-fullheight="385" data-fullwidth="543" height="385" width="543" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr1.sml" data-fulleid="1-s2.0-S0167865510000954-gr1.jpg"><img class="figure large" border="0" alt="Variable importance sensitivity to n and p (toys data)." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr1.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr1.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr1.jpg" data-fullEID="1-s2.0-S0167865510000954-gr1.jpg"></noscript></a></dt><dd id="labelCaptionfig1"><div class="caption">Fig. 1.&nbsp;<p id="">Variable importance sensitivity to <em>n</em> and <em>p</em> (toys data).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr1.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=8f8960b9810d03c8e002c25f3c45e86d" data-label="labelCaptionfig1" data-ref="" data-format="S0167865510000954" href="#gr1" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr1_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=ed27439ae5948a3c76cf526ff9c8ef03" data-label="labelCaptionfig1" data-ref="" data-format="S0167865510000954" href="#gr1" title="View high quality image - Opens in new window">Download high-quality image (306 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr1.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">After simulation, the obtained variables are finally standardized.</p><div class="enunciation"><p><strong class="boldFont">Remark&nbsp;2.1.
                    </strong></p><p>Variable importance is computed conditionally to a given realization even for simulated datasets. This choice which is criticizable if the objective is to reach a good estimation of an underlying constant, is consistent with the idea of staying as close as possible to the experimental situation dealing with a given dataset.</p><p></p></div><p></p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl2.1">2.1. Sensitivity to <em>n</em> and <em>p</em></h3><p class="svArticle section clear" id=""><span id="bfig1"><a href="#fig1" id="ancbfig1" class="intra_ref">Fig. 1</a></span> illustrates the behavior of variable importance for several values of <em>n</em> and <em>p</em>. Parameters <em>ntree</em> and <em>mtry &nbsp;</em> are set to their default values (<span id="mmlsi59" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si59.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=2fcee6abde6dcfd1a867f29ec3c9a61e" title="Click to view the MathML source"><em>ntree</em>=500</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si59.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>500</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi60" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si60.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e816967dd96a580600424726a20ceafb"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si60.gif" data-inlimgeid="1-s2.0-S0167865510000954-si60.gif" height="17" width="75"><noscript><img height="17" border="0" style="vertical-align:bottom" width="75" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si60.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si60.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><msqrt><mrow><mi>p</mi></mrow></msqrt></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for the classification case). Boxplots are based on 50 runs of the RF algorithm and for visibility, we plot the variable importance only for a few variables.</p><!--VALIDHTML--><p class="svArticle section clear" id="">On each row, the first plot is the reference one for which we observe a convenient picture of the relative importance of the initial variables. Then, when <em>p</em> increases tremendously, we try to check if: (1) the situation between the two groups remains readable; (2) the situation within each group is stable; (3) the importance of the additional noise variables is close to 0.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The situation <span id="mmlsi61" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si61.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=5faf9c4dfdd70809b189fcff1804d4f3" title="Click to view the MathML source">n=500</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si61.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>500</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> (graphs at the top of the figure) corresponds to an “easy” case, where a lot of data are available and <span id="mmlsi62" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si62.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=9cf5cf5f5da6f89dd542ec26202c9e91" title="Click to view the MathML source">n=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si62.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> (graphs at the bottom) to a harder one. For each value of <em>n</em>, three values of <em>p &nbsp;</em> are considered: 6,200 and 500. When <span id="mmlsi63" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si63.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=859df84883cfb29e04afe0a4ea088898" title="Click to view the MathML source">p=6</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si63.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> only the six true variables are present. Then two very difficult situations are considered: <span id="mmlsi64" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si64.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e4a499ca5ca173b17e611ba56d771ab8" title="Click to view the MathML source">p=200</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si64.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>200</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> with a lot of noisy variables and <span id="mmlsi65" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si65.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d9b5721c6779fbbe1bdd691abfa6f31d" title="Click to view the MathML source">p=500</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si65.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>500</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is even harder. We consider such very high values for <em>p</em> to mimic the situation of very high dimensional real data considered in the sequel. Graphs are truncated after the 16th variable for readability (importance of noisy variables left are of the same order of magnitude as the last plotted).</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_6"></div></div><div id="frag_6" class="page_fragment" data-fid="6"><p class="svArticle section clear" id="">Let us comment on graphs on the first row <span id="mmlsi66" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si66.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=da92e34d855a8a13594f62d75e372408" title="Click to view the MathML source">(n=500)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si66.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>500</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. When <span id="mmlsi67" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si67.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ad30f6f48ec2de86df53f96962fc016a" title="Click to view the MathML source">p=6</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si67.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> we obtain concentrated boxplots and the order is clear, variables 2 and 6 having nearly the same importance. When <em>p</em> increases, the order of magnitude of importance decreases (note that the <em>y &nbsp;</em>-axis scale is different for <span id="mmlsi68" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si68.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=a94320ca6a8b06e49e517290bf8f6f9e" title="Click to view the MathML source">p=6</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si68.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and for <span id="mmlsi69" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si69.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e966b4a11403ad94208f1b48d83544e1"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si69.gif" data-inlimgeid="1-s2.0-S0167865510000954-si69.gif" height="16" width="41"><noscript><img height="16" border="0" style="vertical-align:bottom" width="41" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si69.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si69.gif" overflow="scroll"><mrow><mi>p</mi><mspace width="0.25em"></mspace><mo>≠</mo><mspace width="0.25em"></mspace><mn>6</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>). The order within the two groups of variables (1–3 and 4–6) remains the same, while the overall order is modified (variable 6 is now less important than variable 2). In addition, variable importance is more unstable for huge values of <em>p</em>. But what is remarkable is that all noisy variables have a zero VI. So one can easily recover variables of interest.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In the second row <span id="mmlsi70" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si70.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=acd29240800568ef53f56bdd66a16a2c" title="Click to view the MathML source">(n=100)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si70.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>100</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, we note a greater instability since the number of observations is only moderate, but the variable ranking remains quite the same. What differs is that in the difficult situations <span id="mmlsi71" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si71.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=f755d2daf949df133417b5e50a4be564"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si71.gif" data-inlimgeid="1-s2.0-S0167865510000954-si71.gif" height="17" width="102"><noscript><img height="17" border="0" style="vertical-align:bottom" width="102" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si71.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si71.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>p</mi><mo>=</mo><mn>200</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mn>500</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> importance of some noisy variables increases, and for example variable 4 cannot be distinguished from noise. The same holds even for variable 5 for <span id="mmlsi72" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si72.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=42a8ec33858eb0032c70244a77c8f6d6" title="Click to view the MathML source">p=500</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si72.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>500</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. This is due to the decreasing behavior of VI with <em>p &nbsp;</em> growing, coming from the fact that when <span id="mmlsi73" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si73.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ad01a40343f6c1f38ff65036e578e74b" title="Click to view the MathML source">p=500</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si73.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>500</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> the algorithm randomly choose only 22 variables at each split (with the <em>mtry</em> default value). The probability of choosing one of the six true variables is really small and the less a variable is chosen, the less it can be considered as important. We will see the benefits of increasing <em>mtry</em> in the next paragraph.</p><!--VALIDHTML--><p class="svArticle section clear" id="">In addition, it should be noted that the variability of VI is large for true variables with respect to useless ones. This remark can be used to build some kind of test for VI (see <span id="bbib36"><a href="#bib36" id="ancbbib36" class="intra_ref">Strobl et al. (2007)</a></span>) but of course ranking is better suited for variable selection.</p><!--VALIDHTML--><p class="svArticle section clear" id="">We now study how this VI index behaves when changing values of the main method parameters.</p><!--VALIDHTML--><h3 class="svArticle" id="sec2">2.2. Sensitivity to <em>mtry</em> and <em>ntree</em></h3><p class="svArticle section clear" id="">The choice of <em>mtry</em> and <em>ntree &nbsp;</em> can be important for the VI computation. We fix <span id="mmlsi74" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si74.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3d26313ed363d25bec1c96d4302615d1" title="Click to view the MathML source">n=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si74.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi75" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si75.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=6748689d3e5d73a91c7c8c50658fc9d6" title="Click to view the MathML source">p=200</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si75.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>200</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and, in <span id="bfig2"><a href="#fig2" id="ancbfig2" class="intra_ref">Fig. 2</a></span>, we plot variable importance obtained using three values of <em>mtry</em> (14 the default, 100 and 200) and two values of <em>ntree</em> (500 the default, and 2000).</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig2"><div><dl class="figure" id="fig2" data-t="f"><dt class="autoScroll" data-style="height:383px;width:536px;"><a class="figureLink" href="#gr2" title="Variable importance sensitivity to mtry and ntree (toys data)."><img data-loaded="true" style="display: inline; height: 383px; width: 536px;" class="imgLazyJSB figure large nrmImg" alt="Variable importance sensitivity to mtry and ntree (toys data)." src="1-s2.0-S0167865510000954-gr2.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr2.sml" data-fulleid="1-s2.0-S0167865510000954-gr2.jpg" data-imgeids="1-s2.0-S0167865510000954-gr2.jpg" data-thumbheight="157" data-thumbwidth="219" data-fullheight="383" data-fullwidth="536" height="383" width="536" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr2.sml" data-fulleid="1-s2.0-S0167865510000954-gr2.jpg"><img class="figure large" border="0" alt="Variable importance sensitivity to mtry and ntree (toys data)." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr2.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr2.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr2.jpg" data-fullEID="1-s2.0-S0167865510000954-gr2.jpg"></noscript></a></dt><dd id="labelCaptionfig2"><div class="caption">Fig. 2.&nbsp;<p id="">Variable importance sensitivity to <em>mtry</em> and <em>ntree</em> (toys data).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr2.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=0a709ca1da53e3d1f02306a829df1253" data-label="labelCaptionfig2" data-ref="" data-format="S0167865510000954" href="#gr2" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr2_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=4723a0f4bfeb079920d730685ced220a" data-label="labelCaptionfig2" data-ref="" data-format="S0167865510000954" href="#gr2" title="View high quality image - Opens in new window">Download high-quality image (341 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr2.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">The effect of taking a larger value for <em>mtry &nbsp;</em> is obvious. Indeed the magnitude of VI is more than doubled starting from <span id="mmlsi76" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si76.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=178d80fa87e81cc48e00f260f15b05ed" title="Click to view the MathML source"><em>mtry</em>=14</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si76.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mn>14</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> to <span id="mmlsi77" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si77.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=4c56da47ebcf48507e58e1bedbef3322" title="Click to view the MathML source"><em>mtry</em>=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si77.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, and it again increases whith <span id="mmlsi78" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si78.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c44e864ce7bfaeda57e919287bc6111b" title="Click to view the MathML source"><em>mtry</em>=200</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si78.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mn>200</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. The effect of <em>ntree &nbsp;</em> is less visible, but taking <span id="mmlsi79" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si79.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=b6b396e7b87e4fe5cb62595328e683f6" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si79.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> leads to better stability. What is difficult to see but interesting in the bottom right graph is that we get the same order for all true variables in every run of the procedure.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_7"></div></div><div id="frag_7" class="page_fragment" data-fid="7"><h3 class="svArticle" id="section_lbl2.3">2.3. Sensitivity to highly correlated predictors</h3><p class="svArticle section clear" id="">We now illustrate an important issue: how does variable importance behave in presence of several highly correlated variables? We take as basic framework the previous context with <span id="mmlsi80" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si80.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d40b811c6af77b9c71f4f491dd165d5e"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si80.gif" data-inlimgeid="1-s2.0-S0167865510000954-si80.gif" height="16" width="221"><noscript><img height="16" border="0" style="vertical-align:bottom" width="221" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si80.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si80.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>100</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi>p</mi><mo>=</mo><mn>200</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi81" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si81.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=5167d29c3b15bc6efacdd1f992810d79" title="Click to view the MathML source"><em>mtry</em>=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si81.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. Then we add to the dataset highly correlated replications of some of the six true variables.</p><!--VALIDHTML--><p class="svArticle section clear" id="">The first graph of <span id="bfig3"><a href="#fig3" id="ancbfig3" class="intra_ref">Fig. 3</a></span> is the reference one: the situation is the same as previously. Then for the three other cases, we simulate 1, 10 and 20 variables with a correlation of 0.9 with variable 3 (the most important one). These replications are plotted between the two vertical lines.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig3"><div><dl class="figure" id="fig3" data-t="f"><dt class="autoScroll" data-style="height:384px;width:510px;"><a class="figureLink" href="#gr3" title="Variable importance in presence of a group of correlated variables (augmented ..."><img data-loaded="true" style="display: inline; height: 384px; width: 510px;" class="imgLazyJSB figure large nrmImg" alt="Variable importance in presence of a group of correlated variables (augmented ..." src="1-s2.0-S0167865510000954-gr3.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr3.sml" data-fulleid="1-s2.0-S0167865510000954-gr3.jpg" data-imgeids="1-s2.0-S0167865510000954-gr3.jpg" data-thumbheight="164" data-thumbwidth="217" data-fullheight="384" data-fullwidth="510" height="384" width="510" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr3.sml" data-fulleid="1-s2.0-S0167865510000954-gr3.jpg"><img class="figure large" border="0" alt="Variable importance in presence of a group of correlated variables (augmented ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr3.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr3.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr3.jpg" data-fullEID="1-s2.0-S0167865510000954-gr3.jpg"></noscript></a></dt><dd id="labelCaptionfig3"><div class="caption">Fig. 3.&nbsp;<p id="">Variable importance in presence of a group of correlated variables (augmented toys data).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr3.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=3fda5f5dd86c12684e4b2ec1b5b8d77d" data-label="labelCaptionfig3" data-ref="" data-format="S0167865510000954" href="#gr3" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr3_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=27f8c9f347297f2832b5a44d0d89a2a7" data-label="labelCaptionfig3" data-ref="" data-format="S0167865510000954" href="#gr3" title="View high quality image - Opens in new window">Download high-quality image (213 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr3.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">VIs in the group 1,2,3 are steadily decreasing when adding more replications of variable 3. On the other hand, VIs in the group 4,5,6 are unchanged. Notice that the importance is not divided by the number of replications. Indeed in our example, even with 20 replications the maximum of VIs in the group containing variable 3 (that is variables 1 to 3 and all replications of variable 3) is only three times lower than the initial VI of variable 3. Finally, note that even if some variables in this group have low importance, they cannot be confused with noise.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Let us briefly comment on similar experiments (see <span id="bfig4"><a href="#fig4" id="ancbfig4" class="intra_ref">Fig. 4</a></span>) obtained by perturbing the basic situation not only by introducing highly correlated versions of the third variable but also of the sixth, leading to replicate the most important of each group.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig4"><div><dl class="figure" id="fig4" data-t="f"><dt class="autoScroll" data-style="height:384px;width:511px;"><a class="figureLink" href="#gr4" title="Variable importance in presence of two groups of correlated variables (augmented ..."><img data-loaded="true" style="display: inline; height: 384px; width: 511px;" class="imgLazyJSB figure large nrmImg" alt="Variable importance in presence of two groups of correlated variables (augmented ..." src="1-s2.0-S0167865510000954-gr4.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr4.sml" data-fulleid="1-s2.0-S0167865510000954-gr4.jpg" data-imgeids="1-s2.0-S0167865510000954-gr4.jpg" data-thumbheight="164" data-thumbwidth="218" data-fullheight="384" data-fullwidth="511" height="384" width="511" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr4.sml" data-fulleid="1-s2.0-S0167865510000954-gr4.jpg"><img class="figure large" border="0" alt="Variable importance in presence of two groups of correlated variables (augmented ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr4.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr4.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr4.jpg" data-fullEID="1-s2.0-S0167865510000954-gr4.jpg"></noscript></a></dt><dd id="labelCaptionfig4"><div class="caption">Fig. 4.&nbsp;<p id="">Variable importance in presence of two groups of correlated variables (augmented toys data).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr4.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=f7979fdef66ca567e3003445bf41fb72" data-label="labelCaptionfig4" data-ref="" data-format="S0167865510000954" href="#gr4" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr4_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=cdef22e054ad805650753e6eef2964b1" data-label="labelCaptionfig4" data-ref="" data-format="S0167865510000954" href="#gr4" title="View high quality image - Opens in new window">Download high-quality image (220 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr4.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">Again, the first graph is the reference one. Then for the three other cases, we simulate, in each group (1,&nbsp;2,&nbsp;3 and 4,&nbsp;5,&nbsp;6 respectively) 1, 5 and 10 variables of correlation about 0.9 with variables 3 and 6 respectively. Replications of variable 3 are plotted between the first vertical line and the dashed line, and replications of variable 6 between the dashed line and the second vertical line.</p><!--VALIDHTML--><p class="svArticle section clear" id="">VIs within each group are steadily decreasing when adding more replications. Nevertheless, the relative position between the two groups is preserved.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl2.4">2.4. Prostate data variable importance</h3><p class="svArticle section clear" id="">To end this section, we illustrate the behavior of variable importance on a high dimensional real dataset: the microarray data called Prostate, for which <span id="mmlsi82" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si82.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d534bf4ba9deee8e72810dd9be70f254" title="Click to view the MathML source">n=102</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si82.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>102</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi83" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si83.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=2bfca14b9010d9992f3f9832002c16f4" title="Click to view the MathML source">p=6033</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si83.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6033</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> (see <span id="bbib34"><a href="#bib34" id="ancbbib34" class="intra_ref">Singh et al. (2002)</a></span> for a detailed presentation). The global picture is the following: two hugely important variables, about twenty moderately important variables and the others of small importance. So, more precisely, <span id="bfig5"><a href="#fig5" id="ancbfig5" class="intra_ref">Fig. 5</a></span> compares VI obtained for parameters set to their default values (graphs of the left column) and those obtained for <span id="mmlsi84" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si84.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3e20b7df8cca3ff2ae8f3e91dae79bc0" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si84.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi85" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si85.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=501d07a863d955c486981bcaca967816" title="Click to view the MathML source"><em>mtry</em>=p/3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si85.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> (graphs of the right column). Graphs are truncated after the 250th variable for readability (importance of noisy variables left are of the same order of magnitude as the last plotted).</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_8"></div></div><div id="frag_8" class="page_fragment" data-fid="8"><div class="figTblUpiOuter svArticle" id="figure_fig5"><div><dl class="figure" id="fig5" data-t="f"><dt class="autoScroll" data-style="height:474px;width:514px;"><a class="figureLink" href="#gr5" title="Variable importance for Prostate data (using ntree=2000 and mtry=p/3, on the ..."><img data-loaded="true" style="display: inline; height: 474px; width: 514px;" class="imgLazyJSB figure large nrmImg" alt="Variable importance for Prostate data (using ntree=2000 and mtry=p/3, on the ..." src="1-s2.0-S0167865510000954-gr5.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr5.sml" data-fulleid="1-s2.0-S0167865510000954-gr5.jpg" data-imgeids="1-s2.0-S0167865510000954-gr5.jpg" data-thumbheight="164" data-thumbwidth="178" data-fullheight="474" data-fullwidth="514" height="474" width="514" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr5.sml" data-fulleid="1-s2.0-S0167865510000954-gr5.jpg"><img class="figure large" border="0" alt="Variable importance for Prostate data (using ntree=2000 and mtry=p/3, on the ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr5.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr5.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr5.jpg" data-fullEID="1-s2.0-S0167865510000954-gr5.jpg"></noscript></a></dt><dd id="labelCaptionfig5"><div class="caption">Fig. 5.&nbsp;<p id="">Variable importance for Prostate data (using <span id="mmlsi1" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si1.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=8b4fcbed07132fb796249e326727c9b2" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si1.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi2" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si2.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3df27b8c3d5cf2455c3a4e1363b8b2b4" title="Click to view the MathML source"><em>mtry</em>=p/3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si2.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, on the right and using default values on the left).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr5.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=03c808bd085b6052198ee49195878723" data-label="labelCaptionfig5" data-ref="" data-format="S0167865510000954" href="#gr5" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr5_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=88a0207a3467c10eeda1cd40b2deaa04" data-label="labelCaptionfig5" data-ref="" data-format="S0167865510000954" href="#gr5" title="View high quality image - Opens in new window">Download high-quality image (380 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr5.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">For the two most important variables (first row), the magnitude of importance obtained with <span id="mmlsi86" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si86.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=297849e8041608c89020d85e0774fa68" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si86.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi87" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si87.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=300d6f0b81623456b433619e36ee6ccf" title="Click to view the MathML source"><em>mtry</em>=p/3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si87.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is much larger than to the one obtained with default values. In the second row, the increase of magnitude is still noticeable from the third to the 9th most important variables and from the 10th to the 20th most important variables, VI is quite the same for the two parameter choices. In the third row, we get VI closer to zero for the variables with <span id="mmlsi88" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si88.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=635d58981cb64e36e194e9011ec7d360" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si88.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi89" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si89.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=902cd6496fefc42f05cbd6e07a3630f1" title="Click to view the MathML source"><em>mtry</em>=p/3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si89.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> than with default values. In addition, note that for the less important variables, boxplots are larger for default values, especially for unimportant variables (from the 200th to the 250th).</p><!--VALIDHTML--><h2 class="svArticle" id="sec3">3. Variable selection</h2><p class="svArticle section clear" id="">We distinguish two variable selection objectives:</p><dl class="listitem" id="list_"><dt class="label">1.</dt><dd><p id="">To find important variables highly related to the response variable for interpretation purpose;</p></dd><dt class="label">2.</dt><dd><p id="">To find a small number of variables sufficient to a good parsimonious prediction of the response variable.</p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">The first one is to magnify all the important variables, even with high redundancy, for interpretation purpose and the second one is to find a sufficient parsimonious set of important variables for prediction.</p><!--VALIDHTML--><p class="svArticle section clear" id="">As mentioned at the end of the introduction, we are guided in this paper by a typical situation matching two characteristics. The first one is high dimensionality, or at least when the number of true variables is much less than <em>p</em>, and the second one is the presence of groups of highly correlated predictors. They are also specifically addressed in two earlier works by  <a href="#bib14" id="bbib14" class="intra_ref">Díaz-Uriarte and Alvarez de Andrés, 2006</a>&nbsp;and&nbsp;<a href="#bib4" id="bbib4" class="intra_ref">Ben Ishak and Ghattas, 2008</a>. We briefly recall these contributions.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Díaz-Uriarte, Alvarez de Andrés propose a strategy based on recursive elimination of variables. More precisely, they first compute RF variable importance. Then, at each step, they eliminate the 20% of the variables having the smallest importance and build a new forest with the remaining variables. They finally select the set of variables leading to the smallest OOB error rate of a forest, defined by</p><div class="formula" id=""><div class="mathml"><span id="mmlsi90" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si90.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=608b9af02ff7e5bf6a53210152d9a857"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si90.gif" data-inlimgeid="1-s2.0-S0167865510000954-si90.gif" height="37" width="293"><noscript><img height="37" border="0" style="vertical-align:bottom" width="293" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si90.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si90.gif" overflow="scroll"><mi mathvariant="italic">errOOB</mi><mo>=</mo><mfrac><mrow><mn>1</mn></mrow><mrow><mi>n</mi></mrow></mfrac><mi mathvariant="italic">Card</mi><mo stretchy="false">{</mo><mi>i</mi><mo>∈</mo><mo stretchy="false">{</mo><mn>1</mn><mtext>,</mtext><mo>…</mo><mtext>,</mtext><mi>n</mi><mo stretchy="false">}</mo><mo stretchy="false">|</mo><msub><mrow><mi>y</mi></mrow><mrow><mi>i</mi></mrow></msub><mspace width="0.25em"></mspace><mo>≠</mo><mspace width="0.25em"></mspace><msub><mrow><mover accent="true"><mrow><mi>y</mi></mrow><mrow><mo>ˆ</mo></mrow></mover></mrow><mrow><mi>i</mi></mrow></msub><mo stretchy="false">}</mo><mtext>,</mtext></math></span><!--mathContainer--></span><div class="mathjax"><a href="javascript:void(0);" role="button" class="mathButton mathjax" title="Turn MathJax on" aria-hidden="true">Turn<span class="mathjax ">&nbsp;</span><span class="offscreen">MathJax </span>on</a></div><!--Loading Mathjax--></span><img class="temp" src="blank.gif"><!--Comment--></div></div>where <span id="mmlsi91" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si91.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=746a0e8b195aef8021a89d8c904fad44"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si91.gif" data-inlimgeid="1-s2.0-S0167865510000954-si91.gif" height="16" width="13"><noscript><img height="16" border="0" style="vertical-align:bottom" width="13" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si91.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si91.gif" overflow="scroll"><mrow><msub><mrow><mover accent="true"><mrow><mi>y</mi></mrow><mrow><mo>ˆ</mo></mrow></mover></mrow><mrow><mi>i</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is the most frequent label predicted by trees <em>t &nbsp;</em> for which <span id="mmlsi92" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si92.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=427d2dcdc01f9751cd86ef8611086719"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si92.gif" data-inlimgeid="1-s2.0-S0167865510000954-si92.gif" height="17" width="47"><noscript><img height="17" border="0" style="vertical-align:bottom" width="47" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si92.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si92.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi></mrow></msub><mtext>,</mtext><mspace width="0.35em"></mspace><msub><mrow><mi>y</mi></mrow><mrow><mi>i</mi></mrow></msub><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> is in the <span id="mmlsi93" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si93.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=46c53c7ecaea98c283c25d7cf118cc2d" title="Click to view the MathML source"><em>OOB</em><sub>t</sub></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si93.gif" overflow="scroll"><mrow><msub><mrow><mi mathvariant="italic">OOB</mi></mrow><mrow><mi>t</mi></mrow></msub></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> sample. The proportion of variables to eliminate is an arbitrary parameter of their method and does not depend on the data.<p></p><!--VALIDHTML--><p class="svArticle section clear" id="">Ben Ishak, Ghattas choose an ascendant strategy based on a sequential introduction of variables. First, they compute some SVM-based variable importance. Then, they build a sequence of SVM models invoking at the beginning the <em>k</em> most important variables, by step of 1. When <em>k</em> becomes too large, the additional variables are invoked by blocks. They finally select the set of variables leading to the model of smallest error rate. The way to introduce variables is not data-driven since it is fixed before running the procedure. They also compare their procedure with a similar one using RF instead of SVM.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl3.1">3.1. Procedure</h3><p class="svArticle section clear" id="">We propose the following two-steps procedure, the first one is common while the second one depends on the objective:</p><dl class="listitem" id="list_"><dt class="label">Step 1.</dt><dd><p id="">Preliminary elimination and ranking:</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">Sort the variables in decreasing order of RF scores of importance.</p></dd><dt class="label">•</dt><dd><p id="">Cancel the variables of small importance. Denote by <em>m</em> the number of remaining variables.</p></dd></dl><p></p></dd><dt class="label">Step 2.</dt><dd><p id="">Variable selection:</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">For <em>interpretation</em>: construct the nested collection of RF models involving the <em>k &nbsp;</em> first variables, for <span id="mmlsi94" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si94.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=88df909959880209c55cb1083df21c7e" title="Click to view the MathML source">k=1</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si94.gif" overflow="scroll"><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> to <em>m</em>, and select the variables involved in the model leading to the smallest OOB error;</p></dd><dt class="label">•</dt><dd><p id="">For <em>prediction</em>: starting from the ordered variables retained for interpretation, construct an ascending sequence of RF models, by invoking and testing the variables stepwise. The variables of the last model are selected.</p></dd></dl><p></p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">Of course, this is a sketch of procedure and more details are needed to be effective. The next paragraph answer this point but we emphasize that we propose an heuristic strategy which does not depend on specific model hypotheses but based on data-driven thresholds to take decisions.</p><div class="enunciation"><p><strong class="boldFont">Remark&nbsp;3.1.
                    </strong></p><p>Since we want to treat in an unified way all the situations, we will use for finding prediction variables the somewhat crude strategy previously defined. Nevertheless, starting from the set of variables selected for interpretation (say of size <em>K &nbsp;</em>), a better strategy could be to examine all, or at least a large part, of the <span id="mmlsi95" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si95.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=82e6663833970ac04019ec2109ccf66c" title="Click to view the MathML source">2<sup>K</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si95.gif" overflow="scroll"><mrow><msup><mrow><mn>2</mn></mrow><mrow><mi>K</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> possible models and to select the variables of the model minimizing the OOB error. But this strategy becomes quickly unrealistic for high dimensional problems so we prefer to experiment a strategy designed for small <em>n</em> and large <em>K</em> which is not conservative and even possibly leads to select fewer variables.</p><p></p></div><p></p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_9"></div></div><div id="frag_9" class="page_fragment" data-fid="9"><h3 class="svArticle" id="section_lbl3.2">3.2. Starting example</h3><p class="svArticle section clear" id="">To both illustrate and give more details about this procedure, we apply it on a simulated learning set of size <span id="mmlsi96" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si96.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e65abe6418a55178e7f8cef4099e6563" title="Click to view the MathML source">n=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si96.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> from the classification toys data model with <span id="mmlsi97" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si97.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e6a81cb15d72d932d55402d59f3ea71c" title="Click to view the MathML source">p=200</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si97.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>200</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. The results are summarized in <span id="bfig6"><a href="#fig6" id="ancbfig6" class="intra_ref">Fig. 6</a></span>. The true variables (1–6) are respectively represented by <span id="mmlsi98" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si98.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=9dcc6fcdf85f502e3aba2bd530b2d534"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si98.gif" data-inlimgeid="1-s2.0-S0167865510000954-si98.gif" height="17" width="139"><noscript><img height="17" border="0" style="vertical-align:bottom" width="139" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si98.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si98.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>▷</mi><mtext>,</mtext><mspace width="0.35em"></mspace><mi>△</mi><mtext>,</mtext><mspace width="0.35em"></mspace><mo>∘</mo><mtext>,</mtext><mspace width="0.35em"></mspace><mi>☆</mi><mtext>,</mtext><mspace width="0.35em"></mspace><mi>◁</mi><mtext>,</mtext><mspace width="0.35em"></mspace><mo>□</mo><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. We compute, thanks to the learning set, 50 forests with <span id="mmlsi99" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si99.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=9e0afb47f16dbd048a5a4b616ba2953b" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si99.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi100" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si100.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=999aec0281f16e634e5b155011a97b80" title="Click to view the MathML source"><em>mtry</em>=100</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si100.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mn>100</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, which are values of the main parameters previously considered as well adapted for VI calculations (see Section <span id="bsec2"><a href="#sec2" id="ancbsec2" class="intra_ref">2.2</a></span>).</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig6"><div><dl class="figure" id="fig6" data-t="f"><dt class="autoScroll" data-style="height:408px;width:541px;"><a class="figureLink" href="#gr6" title="Variable selection procedures for interpretation and prediction for toys data."><img data-loaded="true" style="display: inline; height: 408px; width: 541px;" class="imgLazyJSB figure large nrmImg" alt="Variable selection procedures for interpretation and prediction for toys data." src="1-s2.0-S0167865510000954-gr6.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr6.sml" data-fulleid="1-s2.0-S0167865510000954-gr6.jpg" data-imgeids="1-s2.0-S0167865510000954-gr6.jpg" data-thumbheight="164" data-thumbwidth="217" data-fullheight="408" data-fullwidth="541" height="408" width="541" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr6.sml" data-fulleid="1-s2.0-S0167865510000954-gr6.jpg"><img class="figure large" border="0" alt="Variable selection procedures for interpretation and prediction for toys data." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr6.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr6.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr6.jpg" data-fullEID="1-s2.0-S0167865510000954-gr6.jpg"></noscript></a></dt><dd id="labelCaptionfig6"><div class="caption">Fig. 6.&nbsp;<p id="">Variable selection procedures for interpretation and prediction for toys data.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr6.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=38161d83526f1b46c0374fcb66bfe1a5" data-label="labelCaptionfig6" data-ref="" data-format="S0167865510000954" href="#gr6" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr6_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=c10e34d5d438f5d05f48ec5863b0b087" data-label="labelCaptionfig6" data-ref="" data-format="S0167865510000954" href="#gr6" title="View high quality image - Opens in new window">Download high-quality image (239 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr6.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_10"></div></div><div id="frag_10" class="page_fragment" data-fid="10"><p class="svArticle section clear" id="">Let us detail the main stages of the procedure together with, in italics, the results obtained on toys data:</p><dl class="listitem" id="list_"><dt class="label">•</dt><dd><p id="">Variable ranking. First we rank the variables by sorting the VI (averaged from the 50 runs) in descending order.</p><p id=""><em>The result is drawn on the top left graph for the 50 most important variables (the other noisy variables having an importance very close to zero too). Note that true variables are significantly more important than the noisy ones.</em></p></dd><dt class="label">•</dt><dd><p id="">Variable elimination. We keep this order in mind and plot the corresponding standard deviations of VI. We use this graph to estimate some threshold for importance. More precisely, we set the threshold as the minimum prediction value given by a CART model fitting this curve (see <span id="bfig7"><a href="#fig7" id="ancbfig7" class="intra_ref">Fig. 7</a></span>). Then we keep only the variables with an averaged VI exceeding this level. This rule is, in general, conservative and leads to retain more variables than necessary, in order to make a careful choice later.</p><div class="figTblUpiOuter svArticle" id="figure_fig7"><div><dl class="figure" id="fig7" data-t="f"><dt class="autoScroll" data-style="height:410px;width:561px;"><a class="figureLink" href="#gr7" title="Selecting the threshold for variable elimination using CART. Bold line refers to ..."><img data-loaded="true" style="display: inline; height: 410px; width: 561px;" class="imgLazyJSB figure large nrmImg" alt="Selecting the threshold for variable elimination using CART. Bold line refers to ..." src="1-s2.0-S0167865510000954-gr7.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr7.sml" data-fulleid="1-s2.0-S0167865510000954-gr7.jpg" data-imgeids="1-s2.0-S0167865510000954-gr7.jpg" data-thumbheight="160" data-thumbwidth="219" data-fullheight="410" data-fullwidth="561" height="410" width="561" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr7.sml" data-fulleid="1-s2.0-S0167865510000954-gr7.jpg"><img class="figure large" border="0" alt="Selecting the threshold for variable elimination using CART. Bold line refers to ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr7.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr7.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr7.jpg" data-fullEID="1-s2.0-S0167865510000954-gr7.jpg"></noscript></a></dt><dd id="labelCaptionfig7"><div class="caption">Fig. 7.&nbsp;<p id="">Selecting the threshold for variable elimination using CART. Bold line refers to the CART estimation of the dotted line and the horizontal dashed line indicates the threshold (the bottom graph being a zoom of the top one).</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr7.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=b9bd88a9bed353ea4f154f1357e25cad" data-label="labelCaptionfig7" data-ref="" data-format="S0167865510000954" href="#gr7" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr7_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=361673352b3cb5f69b4daade593c5e0c" data-label="labelCaptionfig7" data-ref="" data-format="S0167865510000954" href="#gr7" title="View high quality image - Opens in new window">Download high-quality image (271 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr7.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><p></p><p id=""><em>The standard deviations of VI can be found in the top graph of</em> <span id="bfig7"><a href="#fig7" id="ancbfig7" class="intra_ref"><em>Fig. 7</em></a></span><em>. We can see that true variables standard deviation is large compared to the noisy variables one, which is close to zero. The threshold leads to retain 33 variables. Note that the threshold value is based on VI standard deviations while the effective thresholding is performed on VI mean (top left graph).</em></p><p id="">Of course, this strategy is sensible when there exist irrelevant variables. Otherwise, a classical alternative is to select the threshold according to some elbow finding strategy on the VI mean curve. Some ideas can be found among those used for selecting the number of principal components in PCA (see <span id="bbib28"><a href="#bib28" id="ancbbib28" class="intra_ref">Jolliffe (2002)</a></span>).</p></dd><dt class="label">•</dt><dd><p id="">Variable selection procedure for interpretation. We compute OOB error rates of random forests (averaged on 50 runs and using default parameters) of the nested models starting from the one with only the most important variable, and ending with the one involving all important variables kept previously. Ideally, the variables of the model leading to the smallest OOB error are selected. In fact, in order to deal with instability, we use a classical trick: we select the smallest model with an OOB error less than the minimal OOB error augmented by its empirical standard deviation (based on 50 runs).</p><p id=""><em>Note that in the bottom left graph the error decreases quickly and reaches its minimum when the first four true variables are included in the model. Then it remains nearly constant. We select the model containing four of the six true variables, while the actual minimum is reached with 24 variables.</em></p></dd><dt class="label">•</dt><dd><p id="">Variable selection procedure for prediction. We perform a sequential variable introduction with testing: a variable is added only if the error gain exceeds a threshold. The idea is that the error decrease must be significantly greater than the average variation obtained by adding noisy variables.</p><p id=""><em>The bottom right graph shows the result of this step, the final model for prediction purpose involves only variables 3, 6 and 5. The threshold is set to the mean of the absolute values of the first order differentiated OOB errors between the model with &nbsp;</em><span id="mmlsi101" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si101.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c6775205077588efdeb9defe3993eaa7" title="Click to view the MathML source">p<sub><em>interp</em></sub>=4</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si101.gif" overflow="scroll"><mrow><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">interp</mi></mrow></msub><mo>=</mo><mn>4</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><em>variables (the model we selected for interpretation, see the bottom left graph) and the one with all the &nbsp;</em><span id="mmlsi102" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si102.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3c49fd189779c72086d756bf0237ec89" title="Click to view the MathML source">p<sub><em>elim</em></sub>=33</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si102.gif" overflow="scroll"><mrow><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">elim</mi></mrow></msub><mo>=</mo><mn>33</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><em>variables:</em></p><div class="formula" id=""><div class="mathml"><span id="mmlsi103" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si103.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=7f90a0ddf79e8b054537647f31aac348"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si103.gif" data-inlimgeid="1-s2.0-S0167865510000954-si103.gif" height="52" width="347"><noscript><img height="52" border="0" style="vertical-align:bottom" width="347" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si103.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si103.gif" overflow="scroll"><mfrac><mrow><mn>1</mn></mrow><mrow><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">elim</mi></mrow></msub><mo>-</mo><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">interp</mi></mrow></msub></mrow></mfrac><munderover><mrow><mo>∑</mo></mrow><mrow><mi>j</mi><mo>=</mo><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">interp</mi></mrow></msub></mrow><mrow><msub><mrow><mi>p</mi></mrow><mrow><mi mathvariant="italic">elim</mi></mrow></msub><mo>-</mo><mn>1</mn></mrow></munderover><mo stretchy="false">|</mo><mi mathvariant="italic">errOOB</mi><mo stretchy="false">(</mo><mi>j</mi><mo>+</mo><mn>1</mn><mo stretchy="false">)</mo><mo>-</mo><mi mathvariant="italic">errOOB</mi><mo stretchy="false">(</mo><mi>j</mi><mo stretchy="false">)</mo><mo stretchy="false">|</mo><mtext>,</mtext></math></span><!--mathContainer--></span><div class="mathjax"><a href="javascript:void(0);" role="button" class="mathButton mathjax" title="Turn MathJax on" aria-hidden="true">Turn<span class="mathjax ">&nbsp;</span><span class="offscreen">MathJax </span>on</a></div><!--Loading Mathjax--></span><img class="temp" src="blank.gif"><!--Comment--></div></div><em>where &nbsp;</em><span id="mmlsi104" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si104.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=9827811bf5120726aa48efacbc40031f" title="Click to view the MathML source"><em>errOOB</em>(j)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si104.gif" overflow="scroll"><mrow><mi mathvariant="italic">errOOB</mi><mo stretchy="false">(</mo><mi>j</mi><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><em>is the OOB error of the RF built using the j most important variables.</em><p></p></dd></dl><p></p><!--VALIDHTML--><p class="svArticle section clear" id="">It should be noted that if one wants to estimate the prediction error, since ranking and selection are made on the same set of observations, of course an error evaluation on a test set or using a cross-validation scheme should be preferred. It is taken into account in the next section when our results are compared to others.</p><!--VALIDHTML--><p class="svArticle section clear" id="">To evaluate fairly the different prediction errors, we prefer here to simulate a test set of the same size than the learning set. The test error rate with all (200) variables is about 6% while the one with the four variables selected for interpretation is about 4.5%, a little bit smaller. The model with prediction variables 3, 6 and 5 reaches an error of 1%. Repeating the global procedure 10 times on the same data always gave the same interpretation set of variables and the same prediction set, in the same order.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl3.3">3.3. Highly correlated variables</h3><p class="svArticle section clear" id="">We now apply the procedure on toys data with replicated variables: a first group of variables highly correlated with variable 3 and a second one replicated from variable 6 (the most important variable of each group). The situations of interest are the same as those considered to produce <span id="bfig4"><a href="#fig4" id="ancbfig4" class="intra_ref">Fig. 4</a></span>.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Let us comment on <span id="btbl1"><a href="#tbl1" id="ancbtbl1" class="intra_ref">Table 1</a></span>, where the expression <span id="mmlsi105" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si105.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=798bc8c1042f3e47d02f31b28f900477" title="Click to view the MathML source">i<sup>j</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si105.gif" overflow="scroll"><mrow><msup><mrow><mi>i</mi></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> means that variable <em>i</em> is a replication of variable <em>j</em>.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="table_tbl1"><div><dl id="tbl1" data-t="t" class="table " data-label="Table 1"><dd class="lblCap"><div class="caption"><span class="label">Table 1.
        </span><p>Variable selection procedures in presence of highly correlated variables (augmented toys data) where the expression <span id="mmlsi3" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si3.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c31004350d1f0c3b7d00c76c7ffb067d" title="Click to view the MathML source">i<sup>j</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si3.gif" overflow="scroll"><mrow><msup><mrow><mi>i</mi></mrow><mrow><mi>j</mi></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> means that variable <em>i</em> is a replication of variable <em>j</em>.</p></div></dd><dd class="table" title="Variable selection procedures in presence of highly correlated variables ..."><table><colgroup><col><col><col></colgroup><thead><tr><th rowspan="1" colspan="1" scope="col">Number of replications</th><th rowspan="1" colspan="1" scope="col">Interpretation set</th><th rowspan="1" colspan="1" scope="col">Prediction set</th></tr></thead><tbody><tr><td rowspan="1" colspan="1" class="alignTDchar">1</td><td rowspan="1" colspan="1" class="alignTD">3 <span id="mmlsi4" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si4.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=1dc8647135ce777ffea9ed426c66bdb2" title="Click to view the MathML source">7<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si4.gif" overflow="scroll"><mrow><msup><mrow><mn>7</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> 2 6 5</td><td rowspan="1" colspan="1" class="alignTD">3 6 5</td></tr><tr><td rowspan="1" colspan="1" class="alignTDchar">5</td><td rowspan="1" colspan="1" class="alignTD">3 2 <span id="mmlsi5" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si5.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c03480d41748e20cbf888d8879c87d24" title="Click to view the MathML source">7<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si5.gif" overflow="scroll"><mrow><msup><mrow><mn>7</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><span id="mmlsi6" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si6.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=cf0d29ca44dd03e6ce8b8029a2571741" title="Click to view the MathML source">10<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si6.gif" overflow="scroll"><mrow><msup><mrow><mn>10</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> 6 <span id="mmlsi7" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si7.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=747b04ac14dbc702156169ad73962f66" title="Click to view the MathML source">11<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si7.gif" overflow="scroll"><mrow><msup><mrow><mn>11</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> 5 <span id="mmlsi8" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si8.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=21253328fe29ed153a45546cc863c2cf" title="Click to view the MathML source">12<sup>6</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si8.gif" overflow="scroll"><mrow><msup><mrow><mn>12</mn></mrow><mrow><mn>6</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span></td><td rowspan="1" colspan="1" class="alignTD">3 6 5</td></tr><tr><td rowspan="1" colspan="1" class="alignTDchar">10</td><td rowspan="1" colspan="1" class="alignTD">3 <span id="mmlsi9" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si9.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=53a64aa404e311febba75d88dd90c893" title="Click to view the MathML source">14<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si9.gif" overflow="scroll"><mrow><msup><mrow><mn>14</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><span id="mmlsi10" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si10.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=efbb32e1e64e60ee711c5c231b931760" title="Click to view the MathML source">8<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si10.gif" overflow="scroll"><mrow><msup><mrow><mn>8</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> 2 <span id="mmlsi11" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si11.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c505d26ffef5ac2caa4cd6c56ba151e5" title="Click to view the MathML source">15<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si11.gif" overflow="scroll"><mrow><msup><mrow><mn>15</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> 6 5 <span id="mmlsi12" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si12.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=9cb6166666d08c21384d517bcfc035ef" title="Click to view the MathML source">10<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si12.gif" overflow="scroll"><mrow><msup><mrow><mn>10</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><span id="mmlsi13" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si13.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=3ca4b7e249dcaafe31ce912a89893b79" title="Click to view the MathML source">13<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si13.gif" overflow="scroll"><mrow><msup><mrow><mn>13</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span><span id="mmlsi14" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si14.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d785c89692079abdbd8cbd12c169257d" title="Click to view the MathML source">20<sup>6</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si14.gif" overflow="scroll"><mrow><msup><mrow><mn>20</mn></mrow><mrow><mn>6</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span></td><td rowspan="1" colspan="1" class="alignTD">3 6 5 <span id="mmlsi15" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si15.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=475b84da6a4d3ea9bc747a499fbb2d9c" title="Click to view the MathML source">10<sup>3</sup></span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si15.gif" overflow="scroll"><mrow><msup><mrow><mn>10</mn></mrow><mrow><mn>3</mn></mrow></msup></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span></td></tr></tbody></table></dd><dd class="fullsizeTable"><a data-url="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=tbl1&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;_isTablePopup=Y&amp;md5=25d80e38f273704e601967384c134248" class="tableAnchor" title="Full-size table - Opens new window" href="#">Full-size table</a></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Table options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a href="#tbl1" class="viewWS S_C_table_view_WS" title="View in workspace">View in workspace</a></li><li><div class="downloadCsv"><a class="S_C_csv_download" title="Download as CSV" href="#">Download as CSV</a></div></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">Interpretation sets do not contain all variables of interest. Particularly we hardly keep replications of variable 6. The reason is that even before adding noisy variables to the model the error rate of nested models do increase (or remain constant): when several highly correlated variables are added, the bias remains the same while the variance increases. However the prediction sets are satisfactory: we always highlight variables 3 and 6 and at most one correlated variable with each of them.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_11"></div></div><div id="frag_11" class="page_fragment" data-fid="11"><p class="svArticle section clear" id="">Even if all the variables of interest do not appear in the interpretation set, they always appear in the first positions of our ranking according to importance. More precisely the 16 most important variables in the case of 5 replications are: <span id="mmlsi106" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si106.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=bffc86f38bd8b6bc4443958726420eef"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si106.gif" data-inlimgeid="1-s2.0-S0167865510000954-si106.gif" height="19" width="346"><noscript><img height="19" border="0" style="vertical-align:bottom" width="346" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si106.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si106.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mn>3</mn><mspace width="0.35em"></mspace><mn>2</mn><mspace width="0.35em"></mspace><msup><mrow><mn>7</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>10</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><mn>6</mn><mspace width="0.35em"></mspace><msup><mrow><mn>11</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><mn>5</mn><mspace width="0.35em"></mspace><msup><mrow><mn>12</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>8</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>13</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>16</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><mn>1</mn><mspace width="0.35em"></mspace><msup><mrow><mn>15</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>14</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>9</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><mn>4</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, and the 26 most important variables in the case of 10 replications are: <span id="mmlsi107" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si107.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=ab80eb73b67a25a510b77784b778b436"><img data-loaded="true" style="display: inline;" class="imgLazyJSB smallImg inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si107.gif" data-inlimgeid="1-s2.0-S0167865510000954-si107.gif" height="19" width="627"><noscript><img height="19" border="0" style="vertical-align:bottom" width="627" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si107.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si107.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mn>3</mn><mspace width="0.35em"></mspace><msup><mrow><mn>14</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>8</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><mn>2</mn><mspace width="0.35em"></mspace><msup><mrow><mn>15</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><mn>6</mn><mspace width="0.35em"></mspace><mn>5</mn><mspace width="0.35em"></mspace><msup><mrow><mn>10</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>13</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>20</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>21</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>11</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>12</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>18</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><mn>1</mn><mspace width="0.35em"></mspace><msup><mrow><mn>24</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>7</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>26</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>23</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>16</mn></mrow><mrow><mn>3</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>25</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>22</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>17</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><msup><mrow><mn>19</mn></mrow><mrow><mn>6</mn></mrow></msup><mspace width="0.35em"></mspace><mn>4</mn><mspace width="0.35em"></mspace><msup><mrow><mn>9</mn></mrow><mrow><mn>3</mn></mrow></msup><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>. Note that the order of the true variables (3&nbsp;2&nbsp;6&nbsp;5&nbsp;1&nbsp;4) is always the same.</p><!--VALIDHTML--><h2 class="svArticle" id="sec4">4. Experimental results</h2><p class="svArticle section clear" id="">In this section we experiment the proposed procedure on four high dimensional classification datasets and then finally we examine the results on a standard regression problem to illustrate the versatility of the procedure.</p><!--VALIDHTML--><h3 class="svArticle" id="sec5">4.1. Prostate data</h3><p class="svArticle section clear" id="">We apply the variable selection procedure on Prostate data (for which <span id="mmlsi108" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si108.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=a4fd5508a67390776c022540777ade51" title="Click to view the MathML source">n=102</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si108.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>102</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi109" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si109.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=dc618541922d5edeb8d9df14e60f13c3" title="Click to view the MathML source">p=6033</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si109.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6033</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, see <span id="bbib34"><a href="#bib34" id="ancbbib34" class="intra_ref">Singh et al. (2002)</a></span>). The graphs of <span id="bfig8"><a href="#fig8" id="ancbfig8" class="intra_ref">Fig. 8</a></span> are obtained as those of <span id="bfig6"><a href="#fig6" id="ancbfig6" class="intra_ref">Fig. 6</a></span>, except that for the RF procedure, we use <span id="mmlsi110" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si110.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=898ed10ef00d2612ac9ecbcf649e1e2b"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si110.gif" data-inlimgeid="1-s2.0-S0167865510000954-si110.gif" height="17" width="175"><noscript><img height="17" border="0" style="vertical-align:bottom" width="175" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si110.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si110.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and for the bottom left graph, we only plot the 100 most important variables for visibility. The procedure leads to the same picture as previously, except for the OOB error rate along the nested models which is less regular. The first point is to notice that the elimination step leads to keep only 270 variables. The key point is that the procedure selects 9 variables for interpretation, and 6 variables for prediction. The number of selected variables is then very much smaller than <span id="mmlsi111" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si111.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=057912ea583ed2b69bf1a2b080ed01fa" title="Click to view the MathML source">p=6033</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si111.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>6033</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig8"><div><dl class="figure" id="fig8" data-t="f"><dt class="autoScroll" data-style="height:408px;width:548px;"><a class="figureLink" href="#gr8" title="Variable selection procedures for interpretation and prediction for Prostate ..."><img data-loaded="true" style="display: inline; height: 408px; width: 548px;" class="imgLazyJSB figure large nrmImg" alt="Variable selection procedures for interpretation and prediction for Prostate ..." src="1-s2.0-S0167865510000954-gr8.jpg" data-thumbeid="1-s2.0-S0167865510000954-gr8.sml" data-fulleid="1-s2.0-S0167865510000954-gr8.jpg" data-imgeids="1-s2.0-S0167865510000954-gr8.jpg" data-thumbheight="163" data-thumbwidth="219" data-fullheight="408" data-fullwidth="548" height="408" width="548" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr8.sml" data-fulleid="1-s2.0-S0167865510000954-gr8.jpg"><img class="figure large" border="0" alt="Variable selection procedures for interpretation and prediction for Prostate ..." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr8.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr8.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr8.jpg" data-fullEID="1-s2.0-S0167865510000954-gr8.jpg"></noscript></a></dt><dd id="labelCaptionfig8"><div class="caption">Fig. 8.&nbsp;<p id="">Variable selection procedures for interpretation and prediction for Prostate data.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr8.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=7639ecc4eeb6b096339fd2d91537076a" data-label="labelCaptionfig8" data-ref="" data-format="S0167865510000954" href="#gr8" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr8_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=69bae622ed41770702072cddc7979f17" data-label="labelCaptionfig8" data-ref="" data-format="S0167865510000954" href="#gr8" title="View high quality image - Opens in new window">Download high-quality image (272 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr8.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">In addition, to examine the variability of the interpretation and prediction sets the global procedure is repeated five times on the entire Prostate dataset. The five prediction sets are very close to each other. The number of prediction variables fluctuates between 6 and 10, and 5 variables appear in all sets. Among the five interpretation sets, 2 are identical and made of 9 variables and the 3 other are made of 25 variables. The 9 variables of the smallest sets are present in all sets and the biggest sets (of size 25) have 23 variables in common.</p><!--VALIDHTML--><p class="svArticle section clear" id="">So, although the sets of variables are not identical for each run of the procedure, the most important variables are included in all of the sets.</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_12"></div></div><div id="frag_12" class="page_fragment" data-fid="12"><h3 class="svArticle" id="sec6">4.2. Four high dimensional classification datasets</h3><p class="svArticle section clear" id="">Let us consider the four well known high dimensional real datasets called Colon <span id="mmlsi112" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si112.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=a3c18f4a24d854bab96f780fbd4f3e63"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si112.gif" data-inlimgeid="1-s2.0-S0167865510000954-si112.gif" height="17" width="132"><noscript><img height="17" border="0" style="vertical-align:bottom" width="132" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si112.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si112.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>62</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi>p</mi><mo>=</mo><mn>2000</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, see <span id="bbib2"><a href="#bib2" id="ancbbib2" class="intra_ref">Alon et al. (1999)</a></span>, Leukemia <span id="mmlsi113" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si113.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=2f2d483f111895fbc890bc631db48e9e"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si113.gif" data-inlimgeid="1-s2.0-S0167865510000954-si113.gif" height="17" width="132"><noscript><img height="17" border="0" style="vertical-align:bottom" width="132" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si113.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si113.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>38</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi>p</mi><mo>=</mo><mn>3051</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, see <span id="bbib22"><a href="#bib22" id="ancbbib22" class="intra_ref">Golub et al. (1999)</a></span>, Lymphoma <span id="mmlsi114" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si114.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=0ffe3f3005d638741472d41e37b91659"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si114.gif" data-inlimgeid="1-s2.0-S0167865510000954-si114.gif" height="17" width="132"><noscript><img height="17" border="0" style="vertical-align:bottom" width="132" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si114.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si114.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>62</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi>p</mi><mo>=</mo><mn>4026</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, see <span id="bbib1"><a href="#bib1" id="ancbbib1" class="intra_ref">Alizadeh (2000)</a></span> and Prostate <span id="mmlsi115" class="mathmlsrc"><a title="View the MathML source" class="mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si115.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=0aa62d975619ebed63f5797db1424b56"><img data-loaded="true" style="display: inline;" class="imgLazyJSB inlineImage" alt="View the MathML source" title="View the MathML source" src="1-s2.0-S0167865510000954-si115.gif" data-inlimgeid="1-s2.0-S0167865510000954-si115.gif" height="17" width="140"><noscript><img height="17" border="0" style="vertical-align:bottom" width="140" alt="View the MathML source" title="View the MathML source" src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-si115.gif"></noscript></a><span class="mathContainer hidden"><span class="mathCode"><math altimg="si115.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>102</mn><mtext>,</mtext><mspace width="0.35em"></mspace><mi>p</mi><mo>=</mo><mn>6033</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>, see <span id="bbib34"><a href="#bib34" id="ancbbib34" class="intra_ref">Singh et al. (2002)</a></span>. We apply the global variable selection procedure on these four benchmark high dimensional real datasets, and we want to get an estimation of prediction error rates. Since these datasets are of small size and in order to be comparable with the results obtained by other authors, we use a 5-fold cross-validation to estimate the error rate. So we split the sample in five stratified parts, each part is successively used as a test set, and the remaining of the data is used as a learning set. Note that the set of variables selected vary from one fold to another. So, we give in <span id="btbl2"><a href="#tbl2" id="ancbtbl2" class="intra_ref">Table 2</a></span> the misclassification error rate, given by the 5-fold cross-validation, for interpretation and prediction sets of variables respectively. The number into brackets is the average number of selected variables. In addition, one can find the original error which stands for the misclassification rate given by the 5-fold cross-validation achieved with random forests using all variables. This error is calculated using the same partition in five parts and again we use <span id="mmlsi116" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si116.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=62787682f74f997055881aee8be23dd5" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si116.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi117" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si117.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=d03e8acb0f5a913b21b99f9175760b67" title="Click to view the MathML source"><em>mtry</em>=p/3</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si117.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> for all datasets.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="table_tbl2"><div><dl id="tbl2" data-t="t" class="table " data-label="Table 2"><dd class="lblCap"><div class="caption"><span class="label">Table 2.
        </span><p>Variable selection procedure for four high dimensional real datasets. CV-error rate and into brackets the average number of selected variables.</p></div></dd><dd class="table" title="Variable selection procedure for four high dimensional real datasets. CV-error ..."><table><colgroup><col><col><col><col></colgroup><thead><tr><th rowspan="1" colspan="1" scope="col">Dataset</th><th rowspan="1" colspan="1" scope="col">Interpretation</th><th rowspan="1" colspan="1" scope="col">Prediction</th><th rowspan="1" colspan="1" scope="col">Original</th></tr></thead><tbody><tr><td rowspan="1" colspan="1" class="alignTD">Colon</td><td rowspan="1" colspan="1" class="alignTD">0.16 (35)</td><td rowspan="1" colspan="1" class="alignTD">0.20 (8)</td><td rowspan="1" colspan="1" class="alignTDchar">0.14</td></tr><tr><td rowspan="1" colspan="1" class="alignTD">Leukemia</td><td rowspan="1" colspan="1" class="alignTD">0 (1)</td><td rowspan="1" colspan="1" class="alignTD">0 (1)</td><td rowspan="1" colspan="1" class="alignTDchar">0.02</td></tr><tr><td rowspan="1" colspan="1" class="alignTD">Lymphoma</td><td rowspan="1" colspan="1" class="alignTD">0.08 (77)</td><td rowspan="1" colspan="1" class="alignTD">0.09 (12)</td><td rowspan="1" colspan="1" class="alignTDchar">0.10</td></tr><tr><td rowspan="1" colspan="1" class="alignTD">Prostate</td><td rowspan="1" colspan="1" class="alignTD">0.085 (33)</td><td rowspan="1" colspan="1" class="alignTD">0.075 (8)</td><td rowspan="1" colspan="1" class="alignTDchar">0.07</td></tr></tbody></table></dd><dd class="fullsizeTable"><a data-url="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=tbl2&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;_isTablePopup=Y&amp;md5=a85204971b8803a6f03774c80d5a3490" class="tableAnchor" title="Full-size table - Opens new window" href="#">Full-size table</a></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Table options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a href="#tbl2" class="viewWS S_C_table_view_WS" title="View in workspace">View in workspace</a></li><li><div class="downloadCsv"><a class="S_C_csv_download" title="Download as CSV" href="#">Download as CSV</a></div></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">The number of interpretation variables is hugely smaller than <em>p</em>, at most tens to be compared to thousands. The number of prediction variables is very small (always smaller than 12) and the reduction can be very important with respect to the interpretation set size. The errors for the two variable selection procedures are of the same order of magnitude as the original error (but a little bit larger).</p><!--VALIDHTML--><p class="svArticle section clear" id="">We compare these results with the results obtained by Ben Ishak, Ghattas (see tables 9 and 11 in <span id="bbib4"><a href="#bib4" id="ancbbib4" class="intra_ref">Ben Ishak and Ghattas (2008)</a></span>) which have compared their method with five competitors (mentioned in the introduction) for classification problems on these four datasets. Error rates are comparable. With the prediction procedure we always select fewer variables than their procedures (except for their method GLMpath which select less than three variables for all datasets).</p><!--VALIDHTML--><p class="svArticle section clear" id="">One can notice that the results for the dataset Prostate differ from Section <span id="bsec5"><a href="#sec5" id="ancbsec5" class="intra_ref">4.1</a></span> to Section <span id="bsec6"><a href="#sec6" id="ancbsec6" class="intra_ref">4.2</a></span>. This difference can mainly be explained by the use of 5-fold cross-validation in Section <span id="bsec6"><a href="#sec6" id="ancbsec6" class="intra_ref">4.2</a></span>. Indeed the fact that <em>n &nbsp;</em> is very small <span id="mmlsi118" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si118.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=7d6205efeb8e6137f64c7557c39dd6d4" title="Click to view the MathML source">(n=62)</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si118.gif" overflow="scroll"><mrow><mo stretchy="false">(</mo><mi>n</mi><mo>=</mo><mn>62</mn><mo stretchy="false">)</mo></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> makes the method quite unstable with respect to resampling.</p><!--VALIDHTML--><h3 class="svArticle" id="section_lbl4.3">4.3. Ozone data</h3><p class="svArticle section clear" id="">Before ending the paper, we consider a standard regression dataset. Since it is far from matching the two main characteristics which have guided the algorithm principle, it allows us to check that it still work well. We apply the entire procedure to the easy to interpret ozone dataset (it can be retrieved from the R package <tt>mlbench</tt> and detailed information can be found in the corresponding description file). It consists of <span id="mmlsi119" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si119.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=e5120e89658f3a5141b0243a3bae53f4" title="Click to view the MathML source">n=366</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si119.gif" overflow="scroll"><mrow><mi>n</mi><mo>=</mo><mn>366</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> observations of the daily maximum one-hour-average ozone together with <span id="mmlsi120" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si120.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=80ff81347f37314a6507652b95e7a1ba" title="Click to view the MathML source">p=12</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si120.gif" overflow="scroll"><mrow><mi>p</mi><mo>=</mo><mn>12</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> meteorologic explanatory variables. Let us first examine, in <span id="bfig9"><a href="#fig9" id="ancbfig9" class="intra_ref">Fig. 9</a></span> the VI obtained with RF procedure using <span id="mmlsi121" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si121.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=336125b238d822637b54731591829583" title="Click to view the MathML source"><em>mtry</em>=p/3=4</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si121.gif" overflow="scroll"><mrow><mi mathvariant="italic">mtry</mi><mo>=</mo><mi>p</mi><mo>/</mo><mn>3</mn><mo>=</mo><mn>4</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span> and <span id="mmlsi122" class="mathmlsrc"><span class="formulatext stixSupport mathImg" data-mathurl="/science?_ob=MathURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_mathId=si122.gif&amp;_user=111111111&amp;_pii=S0167865510000954&amp;_rdoc=1&amp;_issn=01678655&amp;md5=c2df3f24f219cf1323fe0c8618a8bacc" title="Click to view the MathML source"><em>ntree</em>=2000</span><span class="mathContainer hidden"><span class="mathCode"><math altimg="si122.gif" overflow="scroll"><mrow><mi mathvariant="italic">ntree</mi><mo>=</mo><mn>2000</mn></mrow></math></span><!--mathContainer--></span><!--Loading Mathjax--></span>.</p><!--VALIDHTML--><div class="figTblUpiOuter svArticle" id="figure_fig9"><div><dl class="figure" id="fig9" data-t="f"><dt class="autoScroll" data-style="height:279px;width:369px;"><a class="figureLink" href="#gr9" title="Variable importance for ozone data."><img style="display: inline;" class="imgLazyJSB figure large nrmImg" alt="Variable importance for ozone data." src="grey_pxl.gif" data-thumbeid="1-s2.0-S0167865510000954-gr9.sml" data-fulleid="1-s2.0-S0167865510000954-gr9.jpg" data-imgeids="1-s2.0-S0167865510000954-gr9.jpg" data-thumbheight="164" data-thumbwidth="217" data-fullheight="279" data-fullwidth="369" height="279" width="369" border="0"><noscript class="figure large" border="0" alt="Full-size image" data-thumbeid="1-s2.0-S0167865510000954-gr9.sml" data-fulleid="1-s2.0-S0167865510000954-gr9.jpg"><img class="figure large" border="0" alt="Variable importance for ozone data." src="http://origin-ars.els-cdn.com/content/image/1-s2.0-S0167865510000954-gr9.jpg" data-thumbEID="1-s2.0-S0167865510000954-gr9.sml" data-imgEIDs="1-s2.0-S0167865510000954-gr9.jpg" data-fullEID="1-s2.0-S0167865510000954-gr9.jpg"></noscript></a></dt><dd id="labelCaptionfig9"><div class="caption">Fig. 9.&nbsp;<p id="">Variable importance for ozone data.</p><!--caption--></div></dd><dd class="menuButtonLinks"><div class="btnHolder"><a class="menuTitle" href="#">Figure options</a><div class="down_Btn"></div></div><ul class="menuButton"><li><a class="S_C_full_size" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr9.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=9f752b776213dabf2c46383226eb083b" data-label="labelCaptionfig9" data-ref="" data-format="S0167865510000954" href="#gr9" title="View full-size image - Opens in new window">Download full-size image</a></li><li><a class="S_C_high_quality" data-src="/science?_ob=MiamiCaptionURL&amp;_method=retrieve&amp;_eid=1-s2.0-S0167865510000954&amp;_image=1-s2.0-S0167865510000954-gr9_lrg.jpg&amp;_cid=271524&amp;_explode=defaultEXP_LIST&amp;_idxType=defaultREF_WORK_INDEX_TYPE&amp;_alpha=defaultALPHA&amp;_ba=&amp;_rdoc=1&amp;_fmt=FULL&amp;_isHiQual=Y&amp;_issn=01678655&amp;_pii=S0167865510000954&amp;md5=ab01a9180646997086d04c062e1d9369" data-label="labelCaptionfig9" data-ref="" data-format="S0167865510000954" href="#gr9" title="View high quality image - Opens in new window">Download high-quality image (71 K)</a></li><li><a class="ppt S_C_ppt" title="Download as PowerPoint slide" href="http://services.elsevier.com/SDWebExport/export/figure/S0167865510000954/1-s2.0-S0167865510000954-gr9.jpg/ppt">Download as PowerPoint slide</a></li></ul></dd></dl></div></div><!--VALIDHTML--><p class="svArticle section clear" id="">From the left to the right, the 12 explanatory variables are 1-Month, 2-Day of month, 3-Day of week, 5-Pressure height, 6-Wind speed, 7-Humidity, 8-Temperature (Sandburg), 9-Temperature (El Monte), 10-Inversion base height, 11-Pressure gradient, 12-Inversion base temperature, 13-Visibility. Variables are numbered exactly as in <tt>mlbench</tt>, so the 4th variable is the response one.</p><!--VALIDHTML--><p class="svArticle section clear" id="">Three very sensible groups of variables appear from the most to the least important. First, the two temperatures (8 and 9), the inversion base temperature (12) known to be the best ozone predictors, and the month (1), which is an important predictor since ozone concentration exhibits an heavy seasonal component. A second group of clearly less important meteorological variables: pressure height (5), humidity (7), inversion base height (10), pressure gradient (11) and visibility (13). Finally three unimportant variables: day of month (2), day of week (3) of course and more surprisingly wind speed (6). This last fact is classical: wind enter in the model only when ozone pollution arises, otherwise wind and pollution are weakly correlated (see for example <span id="bbib12"><a href="#bib12" id="ancbbib12" class="intra_ref">Cheze et al. (2003)</a></span> highlighting this phenomenon using partial estimators).</p><!--VALIDHTML--><p class="svArticle section clear" id="">Let us now examine the results of the selection procedures (see <span id="bfig10"><a href="#fig10" id="ancbfig10" class="intra_ref">Fig. 10</a></span>).</p><!--VALIDHTML--><div class="page_fragment_ind" data-id="frag_13"></div></div>

               

       
   
        
    </div>
    <div id="footer-area-dummy"></div>
    <div id="footer-area" aria-label="product support" role="contentinfo">
      <div id="sdFoot">
      <div id="footer" aria-label="Footer" role="navigation">	
  <div class="bottomArticle"></div>
  <div class="padding">
    <div id="linkArea">
      <div class="padding links">
        <div style="float: left; width: 0px; height: 1px;"></div>
        <div id="banner-bottom">
          <a class="S_C_aboutElsevier" href="http://www.elsevier.com/" target="_top" title="Elsevier homepage (opens in a new window)"></a>
        </div>
        <ul class="col5">
<li><a target="_top" href="http://www.elsevier.com/solutions/sciencedirect">About ScienceDirect</a></li>
</ul>
<ul class="col5">
<li> <a target="_top" href="http://www.sciencedirect.com/science/contactus">Contact and support</a></li>
</ul>
<ul class="col5">
<li><a target="_top" href="http://www.elsevier.com/locate/termsandconditions?utm_source=sciencedirect&amp;utm_medium=link&amp;utm_campaign=terms">Terms and conditions</a></li>
</ul>
<ul class="col5">
<li><a target="_top" href="http://www.elsevier.com/privacypolicy?utm_source=sciencedirect&amp;utm_medium=link&amp;utm_campaign=privacy">Privacy policy</a></li>
</ul>
        <div class="clear"></div>
        <!-- CSAS will be providing the links-->
      </div>
      <div class="padding copyright">
	<hr>
        Copyright © 2016 Elsevier B.V. or its licensors or contributors. ScienceDirect® is a registered trademark of Elsevier B.V.
      </div>
      <div class="padding" style="padding-bottom: 4px; font-size: 11px;">

              Cookies are used by this site. To decline or learn more, visit our <a href="https://www.elsevier.com/solutions/sciencedirect/support/cookies" target="_top">Cookies</a> page.
        </div> 
      
        
          <div class="padding links">
            <div id="goToggleSite"><a class="mLink" site="Mobile" cknm="Site" lnkorg="article" lnkzn="footer" href="#">Switch to Mobile Site</a></div>
          </div>
	  <form id="pgReload" method="get" action="http://www.sciencedirect.com/science/article/pii/S0167865510000954"></form>
        
      
    </div>
  </div>
</div> <!--footer-->

      </div><!--sdFooter-->
    </div><!--footer-area-->
    </div>
 </div>
 <div style="width: 363px;" id="rightPane" class="column" aria-label="Related content" role="complementary">
  <div style="height: 19374px;" id="rightOuter" class="rightOuter">
    <div id="rightInner" class="rightInner">
       <div class="innerPadding">

    <div id="SD_BIB" class="hidden entry separator appBox sgfNoTitleBar sgfNoGadgetBorder"></div>
 

    
    
        <div id="recommend_related_articles" class="entry separator related rendered firstBlk">
            
                <h2><a href="#articles" class="toggle" aria-controls="relArtList" aria-expanded="true"><span class="iconArrow">&nbsp;</span>Recommended articles</a></h2>
             
             <ol id="relArtList"><li><ol class="articles"><li class="artTitle"><a style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" title="A new variable selection approach using Random Forests" href="http://www.sciencedirect.com/science/article/pii/S0167947312003490" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL">A new variable selection approach using Random Forests</a></li><li style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" class="srcTitle">2013, Computational Statistics &amp; Data Analysis<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S0167947312003490/2/issn/01679473" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>A. Hapfelmeier, K. Ulm</li><li class="title">A new variable selection approach using Random Forests</li><li class="fullSrcTitle">Computational Statistics &amp; Data Analysis, Volume 60, April 2013, Pages 50–69</li><li class="docSubType">Original Research Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_top" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S0167947312003490/pdfft?md5=cd983e0c3dde7f09755222584aec1823&amp;pid=1-s2.0-S0167947312003490-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL"> PDF (899 K)</a><div id="entitlement" class="checkEntitlement" style="display: none;"></div></div></li></ul></ol></div></li></ol></li><li><ol class="articles"><li class="artTitle"><a style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" title="Mining data with random forests: A survey and results of new tests" href="http://www.sciencedirect.com/science/article/pii/S0031320310003973" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL">Mining data with random forests: A survey and results of new tests</a></li><li style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" class="srcTitle">2011, Pattern Recognition<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S0031320310003973/2/issn/00313203" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>A. Verikas, A. Gelzinis, M. Bacauskiene</li><li class="title">Mining data with random forests: A survey and results of new tests</li><li class="fullSrcTitle">Pattern Recognition, Volume 44, Issue 2, February 2011, Pages 330–349</li><li class="docSubType">Original Research Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_top" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S0031320310003973/pdfft?md5=fcd89d91eb58edede342bc7e08659b76&amp;pid=1-s2.0-S0031320310003973-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL"> PDF (1095 K)</a><div id="entitlement" class="checkEntitlement" style="display: none;"></div></div></li></ul></ol></div></li></ol></li><li><ol class="articles"><li class="artTitle"><a style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" title="Empirical characterization of random forest variable importance measures" href="http://www.sciencedirect.com/science/article/pii/S0167947307003076" class="cLink S_C_artTitle" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL">Empirical characterization of random forest variable importance measures</a></li><li style="text-overflow: ellipsis; overflow: hidden; white-space: nowrap;" class="srcTitle">2008, Computational Statistics &amp; Data Analysis<div class="moreInfo"><a data-url="http://www.sciencedirect.com/science/more/S0167947307003076/2/issn/01679473" class="toggleMoreInfo">more</a></div></li><li class="showMore hidden"><div class="moreInfoContent"><ol class="details"><li>Kellie J. Archer, Ryan V. Kimes</li><li class="title">Empirical characterization of random forest variable importance measures</li><li class="fullSrcTitle">Computational Statistics &amp; Data Analysis, Volume 52, Issue 4, 10 January 2008, Pages 2249–2260</li><li class="docSubType">Original Research Article</li><div class="abs"></div><ul class="links"><li class="clearfix pdf"><div class="icon_pdf"><a rel="nofollow" title="Download PDF" target="_top" class="cLink pdfRightPaneLink" href="http://www.sciencedirect.com/science/article/pii/S0167947307003076/pdfft?md5=ba958da2127d4e60b223ef14f5d6afa9&amp;pid=1-s2.0-S0167947307003076-main.pdf" querystr="?_origin=article&amp;_zone=citeRelatedArt&amp;panel=citeRelatedArt&amp;_piiSuggestedFrom=S0167865510000954&amp;_srcFr=EPFL"> PDF (212 K)</a><div id="entitlement" class="checkEntitlement" style="display: none;"></div></div></li></ul></ol></div></li></ol></li><li><a href="http://www.sciencedirect.com/science/recommended-articles/S0167865510000954" class="viewMoreArticles cLink">View more articles »</a></li></ol>
        </div><script>prs.rt('scr_rec_art')</script>
    
    
    
        <div id="cited_by_articles" class="entry separator related closed">
            <h2><a href="#articles" class="toggle" aria-controls="relArtList" aria-expanded="false"><span class="iconArrow">&nbsp;</span>Citing articles (<span id="hitcount">165</span>)</a></h2>
            <ol id="citedByList"><li class="note"><p>This article has not been cited.</p></li></ol> 
        </div>
    
    
    
        <div id="SD_CRA" class="hidden entry separator appBox sgfNoTitleBar sgfNoGadgetBorder"></div>  
    
    
        
            <div id="related_book_content" class="entry separator related closed">
                <h2><a href="#articles" class="toggle" aria-controls="relArtList" aria-expanded="false"><span class="iconArrow">&nbsp;</span>Related book content</a></h2>
                <ol id="relBkList"><li class="note" style="margin-left: 15px;"><p>No articles found.</p></li></ol>
            </div>
        
     
     



    
        <div style="height: 165px; display: block;" class="appContainerDummy"></div>
        <div data-top="390" style="top: 56px; max-height: 784px; width: 363px; right: 43.5px;" data-scroll="15696" class="appContainer js_application_content fixedPane">
            <ul class="hidden" id="dataRepHolder"></ul>
            
                <div id="SD_AT1P" class="boxLink sgfNoTitleBar sgfNoGadgetBorder"><div class="containerApplOver invisible" id="SD_AT1P_299100"><div id="remote_iframe_0_userPrefsDialog" style="display: none; z-index: 10; left: 0px; top: 0px; border-width: 0px 1px 1px; border-style: none solid solid; border-color: -moz-use-text-color rgb(205, 240, 164) rgb(205, 240, 164); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background-color: rgb(238, 247, 227); width: 99%; font-weight: bold;" class=""></div><div id="maincontremote_iframe_0" style="border-style: none;" class="containerAppDetails "><iframe id="remote_iframe_0" onload="sgf.killTimer('299100', 'SD_AT1P');" name="remote_iframe_0" src="S0167865510000954_8.html" style="text-align: left;" "="" scrolling="auto" frameborder="no" height="" width="100%"></iframe></div></div><div class="containerApplOver invisible" id="SD_AT1P_301408"><div id="remote_iframe_1_userPrefsDialog" style="display: none; z-index: 10; left: 0px; top: 0px; border-width: 0px 1px 1px; border-style: none solid solid; border-color: -moz-use-text-color rgb(205, 240, 164) rgb(205, 240, 164); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background-color: rgb(238, 247, 227); width: 99%; font-weight: bold;" class=""></div><div id="maincontremote_iframe_1" style="border-style: none;" class="containerAppDetails "><iframe id="remote_iframe_1" onload="sgf.killTimer('301408', 'SD_AT1P');" name="remote_iframe_1" src="S0167865510000954_9.html" style="text-align: left;" "="" scrolling="auto" frameborder="no" height="" width="100%"></iframe></div></div></div>
            

            
                <div id="SD_FTA1" class="boxLink sgfHorToolbar sgfShowMyApps sgfShowMoreLessApps"><div id="sciverse_toolbar" class="sgfSciverseHoriToolBar">	<div class="toolbarfoot">		<div class="toolbarBodyHorizontal" id="toolbarBody">			<div id="toolbar_SD_FTA1" class="sgfInnerToolbar" style="width: 97%; min-height: 155px; text-align: center; margin: 5px auto;"></div>			<div id="showMoreApps">				<a onclick="showMore()" class="plusSign">+</a>				<a title="Show more applications" id="showMore" onclick="showMore()" class="showMoreLessApps">Show more applications</a>			</div>			<div id="showLessApps">				<a onclick="showLess()" class="plusSign">-</a>				<a title="Show fewer applications" id="showFew" onclick="showLess()" class="showMoreLessApps">Show fewer applications</a>			</div>		</div>	</div></div><div style="display: none;" class="sgfToolBarToggleImage">	<div id="sgfSlideBar" class="sgfSideCollapse" onmouseout="sgfToggleToolBarOut();" onmouseover="sgfToggleToolBarOver();" title="undefined"></div>	<div class="clear"></div></div></div>
            
        </div>
    


</div>

<div class="js_workspace_content popup" aria-live="polite" role="region" aria-label="Overlay" aria-describedby="skuip-3">
    <a aria-label="close overlay" title="close overlay" alt="close overlay" href="#" id="buttonClose" role="button" class="closeBt">&nbsp;</a>
    <div class="scrollArea"><div id="workSpaceView"></div></div>
</div>

    </div><!--rightInner-->
  </div><!--rightOuter-->
 </div><!--rightPane-->
</div><!--page-area-->



<div style="clear: both;"></div>

<form name="citationInfo" id="citationInfo" method="POST" action="http://www.sciencedirect.com/science/article/pii/S0167865510000954">
    <input name="art_citation" value="" type="hidden">
</form>


<script src="sdX_1602R3.js"></script>




<script src="sd_1602R3.js"></script>



<script src="pa_1602R3.js"></script>


<script>

var sciverse="empty";
var SDM=new Object;
SDM.tm=new Object;
SDM.tm.ahm='h'



SDM.ep=new Object;
SDM.ep.downloadCsvURL = "http://services.elsevier.com/SDWebExport/export/table/csv";
SDM.ep.downloadActionURL = "http://www.sciencedirect.com/science/downloadActions/keyEvent/S0167865510000954/01678655";
SDM.ep.gigyaLib="//cdn.gigya.com/JS/socialize.js?apiKey=3__Bt7bHrGM8mjKPwBuOVjWyMi678aF6-PewcnoHJJ6UR1fS5t9NG9RBbKHjE0oA6J";
SDM.ep.imgUrlPrefix = "http://ars.els-cdn.com/content/image/";
SDM.ep.thumbimgUrlPrefix = "http://ars.els-cdn.com/content/thumbimage/";
SDM.ep.mendeleyJS = "https://www.mendeley.com/minified/bookmarklet.js";
SDM.ep.prDevice = "http://pa.elsevier.com";
SDM.ep.pru = "http://pa.elsevier.com";
SDM.ep.ref2ref2by = "http://www.sciencedirect.com/science/relLink/ajaxReq?_eid=1-s2.0-S0167865510000954";
SDM.ep.staticImgUrlPrefix = "http://cdn.els-cdn.com/sd";
SDM.ep.urlPrefix = "http://www.sciencedirect.com/science";
SDM.ep.urlQualifier = "science";
SDM.ep.urlScheme = "http://www.sciencedirect.com";
SDM.ep.userActionURL = "http://www.sciencedirect.com/science/UserActions";
 SDM.helpUrlDomain="http://help.sciencedirect.com/flare"; 




SDM.pm=new Object;
SDM.pm.cid = '271524';
SDM.pm.contentFamily = "serial";
SDM.pm.contentType = "JL";
SDM.pm.coverDate = '15-Oct-2010';

SDM.pm.doi = '10.1016/j.patrec.2010.03.014';

SDM.doi = '10.1016/j.patrec.2010.03.014';
SDM.pm.eid = '1-s2.0-S0167865510000954';
SDM.pm.entitled = true;
 SDM.pm.hubEid = '1-s2.0-S0167865510X00125'; 
SDM.pm.indexType = '';
SDM.pm.issnisbn = "01678655";

SDM.pm.itemWeight = 'FULL-TEXT';
SDM.pm.mod_cid="";
SDM.pm.mod_srcTitle="";
SDM.pm.mod_ISBN = "";
SDM.pm.pdfEid = '1-s2.0-S0167865510000954-main.pdf';
SDM.pm.pdfSize = '575130';
SDM.pm.pii = 'S0167865510000954';
SDM.pm.prodColor="sci_dir";if(SDM.pm.prodColor=="sci_dir")SDM.pm.prodColor="";
 SDM.pm.rev = '20150514T050813307592'; 
 SDM.pm.hrev = 'vH.11'; 
 SDM.pm.pptURL = 'http://services.elsevier.com/SDWebExport/export/figure/'; 







SDM.pf = new Object;
SDM.pf.pdfAvail = true;
SDM.pf.reqAllFrags = true;
 SDM.pf.maxElemsAllFrags = 12000
SDM.pf.relatedBkAvail=true;
SDM.pf.mlktAvail = true;
SDM.pf.hpccAvail = true;
SDM.pf.blkSocialLogins=true;


SDM.browserUserMsg=true;


SDM.pf.citedByScAvail = true;
SDM.pf.aetEnabled = false;
 SDM.pf.suggestedArtPDFMaxWait="2000";
 SDM.pf.pdfMenuAvail=true; 


SDM.pf.ads = new Object;
SDM.pf.ads.leftUrl="/science/advertisement/left/num/802227/mainCat/general/cat/general/acct/0663a9eb5160246f83055c65cf6aa49c/isn/01678655/doi/10.1016_j.patrec.2010.03.014/page/article/artOAstatus/N";




SDM.pf.dc = new Object;
SDM.pf.dc.retry = 10;
SDM.pf.dc.retryDelay = 300;
SDM.pf.dc.retryFactor = 1.0;
SDM.pf.dc.respObj = '';


SDM.pf.er = new Object();






SDM.pf.frag = new Object();
SDM.pf.frag.fat = '22f07ee301dcf8ef663b968f7cbad3badcc86470c97ba9cf352528c8459ee24a7e6351f7ad99c0e740872d7c2e8d31350b3a8bd1fd0c2b1babd43a913567ce1a5df44e8279aca8b0e5ea1f15f4611ded39e7778b292883602bbb9b991d53873c0750cb4f55efc6769ed4f7046c54bd528afc21d9b9dad9cccd006f21e27cec720fec7816dfa968a4ac7fe5e0d909d0796838b7fbdaddb58f';
SDM.pf.frag.mapRetry = 20;
SDM.pf.frag.mapRetryDelay = 2000;
SDM.pf.frag.mapRetryFactor = 1.1;
SDM.pf.frag.retry = 5;
SDM.pf.frag.retryDelay = 2000;
SDM.pf.frag.foldFactor = 1.5;
SDM.pf.frag.hashWait = 5000;






SDM.pf.gadget = new Object();
SDM.pf.gadget.displayGadgets=true;
SDM.pf.gadget.sciverseUrlPfx="http://www.sciencedirect.com";
SDM.pf.gadget.sciverseJsonObject='';
SDM.pf.gadget.sciverseGadgetDetailsID = "";
SDM.pf.gadget.svShindigLib="http://sciverse-shindig.elsevier.com/gadgets/js/shindig-container:pubsub-2.js?c=1";
SDM.pf.gadget.svSciverseLib="http://www.sciencedirect.com/gadgetservices/gadgets/v9/js/SciverseShindigService,GadgetToolBar,jquery-outside,jquery.ba-bbq.min,SciverseShindigContainer,ScienceDirectShindigServices.js";



SDM.pf.mrwmod = new Object();
SDM.pf.mrwmod.breadCrumbPathURL = "http://www.sciencedirect.com/science/module/breadCrumbPaths/isbn//pii/S0167865510000954";

SDM.pf.mrwmod.ldrAvail = true;


 


SDM.pf.oa = new Object();








SDM.pf.ref = new Object();
SDM.pf.ref.bs = 20;
SDM.pf.ref.citedByURL = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_updateCitedBy=";
SDM.pf.ref.fd = 300;

SDM.pf.ref.resolveURL = "http://www.sciencedirect.com/science/referenceResolution/ajaxRefResol?_pii=S0167865510000954&_refCnt=38&_docType=article";





SDM.pf.toc=new Object;
SDM.pf.toc.retry = 20;
SDM.pf.toc.retryDelay = 1000;
SDM.pf.toc.retryFactor = 1.09;
SDM.pf.toc.outlineGraphics = false;
SDM.pf.toc.outlineTab = "";

SDM.pf.toc.urlTOCLink="http://www.sciencedirect.com/science/article/pii/S0167865510000954";
SDM.pf.toc.urlTOCLinkQueryStr="?_rdoc=1&_fmt=full&_origin=&md5=22543fb95e34d36391c44b9d9cd72533";

SDM.pf.ddm = new Object();
SDM.pf.ddm.endpoint = "http://services.elsevier.com/SDWebExport/export/pdfdownload/pdf";






SDM.si = new Object();
SDM.si.webUserId = "641661";
SDM.si.sds = "afc13db0385657b7dd1d32c0988df13de2b9ce6541bfcd721b17abc6c9f51b80f083fb75db3a0dea";
SDM.si.euidCookieObject='a5ea17e70c82c5476a9112a1ed7e642d78ad518a597248d5a88316d942da8ffe31a7bc7629b51aac';

SDM.pageTransKey = "02/19/2016 07:23:08 EST#1848_003#5176#62.204.199.202";
SDM.pageType = "article_full";
SDM.tid = "88b9d0bc-d703-11e5-be66-00000aab0f27";
SDM.ud = "641661|<SAInfo><SAId>202</SAId><SAName>MADRID CONSORTIUM</SAName><SANum>S000000202</SANum></SAInfo>|C000034358|51368|IPRANGE";
SDM.ieVersion="8";

SDM.dac = new Object;
SDM.svapi = SVApi;

</script>


<script src="importer-api_1602R3.js.min"></script>


<script>


initializeArticlePage();
</script>




<script type="text/javascript">/*{literal}<![CDATA[*/window.lightningjs||function(c){function g(b,d){d&&(d+=(/\?/.test(d)?"&":"?")+"lv=1");c[b]||function(){var i=window,h=document,j=b,g=h.location.protocol,l="load",k=0;(function(){function b(){a.P(l);a.w=1;c[j]("_load")}c[j]=function(){function m(){m.id=e;return c[j].apply(m,arguments)}var b,e=++k;b=this&&this!=i?this.id||0:0;(a.s=a.s||[]).push([e,b,arguments]);m.then=function(b,c,h){var d=a.fh[e]=a.fh[e]||[],j=a.eh[e]=a.eh[e]||[],f=a.ph[e]=a.ph[e]||[];b&&d.push(b);c&&j.push(c);h&&f.push(h);return m};return m};var a=c[j]._={};a.fh={};a.eh={};a.ph={};a.l=d?d.replace(/^\/\//,(g=="https:"?g:"http:")+"//"):d;a.p={0:+new Date};a.P=function(b){a.p[b]=new Date-a.p[0]};a.w&&b();i.addEventListener?i.addEventListener(l,b,!1):i.attachEvent("on"+l,b);var q=function(){function b(){return["<head></head><",c,' onload="var d=',n,";d.getElementsByTagName('head')[0].",d,"(d.",g,"('script')).",i,"='",a.l,"'\"></",c,">"].join("")}var c="body",e=h[c];if(!e)return setTimeout(q,100);a.P(1);var d="appendChild",g="createElement",i="src",k=h[g]("div"),l=k[d](h[g]("div")),f=h[g]("iframe"),n="document",p;k.style.display="none";e.insertBefore(k,e.firstChild).id=o+"-"+j;f.frameBorder="0";f.id=o+"-frame-"+j;/MSIE[ ]+6/.test(navigator.userAgent)&&(f[i]="javascript:false");f.allowTransparency="true";l[d](f);try{f.contentWindow[n].open()}catch(s){a.domain=h.domain,p="javascript:var d="+n+".open();d.domain='"+h.domain+"';",f[i]=p+"void(0);"}try{var r=f.contentWindow[n];r.write(b());r.close()}catch(t){f[i]=p+'d.write("'+b().replace(/"/g,String.fromCharCode(92)+'"')+'");d.close();'}a.P(2)};a.l&&setTimeout(q,0)})()}();c[b].lv="1";return c[b]}var o="lightningjs",k=window[o]=g(o);k.require=g;k.modules=c}({});
window.usabilla_live = lightningjs.require("usabilla_live", "//w.usabilla.com/14bb862cc5c4.js");
/*]]>{/literal}*/
if(typeof SDM != "undefined") {
  if(SDM.ud) { usabilla_live("data", {"custom": {"ud":SDM.ud} }); }
  if(SDM.pf&&SDM.pf.oa) {
    if(SDM.pf.oa.item) { usabilla_live("data", {"custom": {"open_access":SDM.pf.oa.item} });}
    if(SDM.pf.oa.archive) { usabilla_live("data", {"custom": {"open_archive":SDM.pf.oa.archive} });}
  }
}
</script>


<div id="sdpa"></div>
<script>SDPA.init();</script>
<div id="appBackground"></div>
<div style="position: fixed;" id="appDialog" class="files">
    <div id="header">
        <h1 class="headline">Download PDFs</h1>
        <img id="logo" class="sd-bubbles" src="sciencedirect.svg" onerror="this.src='http://cdn.els-cdn.com/sd/sciencedirect.png'">
        <a class="close" id="closeWindow" href="#"><img src="close-icon.png"></a>
    </div>
    <div id="contentBlk" class="view">
        <div class="help_button" id="ddmMainHelp"><a class="ddmHelp" target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=DM_main.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)">Help</a></div>
        <div class="help_button" id="ddmNewformatHelp"><a class="ddmHelp" target="sdhelp" href="http://www.sciencedirect.com/science?_ob=HelpURL&amp;_file=DM_newformat.htm&amp;_btn=Y&amp;_zone=TopNavBar&amp;_origin=article" title="Help (Opens new window)">Help</a></div>
        <div id="filesTemplate"></div>
        <div id="namingTemplate"></div> 
    </div>
    <div class="footer"></div>
</div>








<img src="idx" style="display: none;" height="0" width="0"><img src="resource" style="display: none;" height="0" width="0"><img src="resource_001.dat" style="display: none;" height="0" width="0"><img src="resource_002.dat" style="display: none;" height="0" width="0"><div style="z-index: 600; bottom: 0px; right: 2%; position: fixed; width: 117px; height: 34px;" aria-hidden="true" tabindex="0" class="usabilla_live_button_container"><iframe title="Visual Feedback Button" data-tags="bottom" scrolling="no" marginheight="0" marginwidth="0" style="width: 117px; height: 34px; margin: 0px; padding: 0px; border: 0px none; overflow: hidden; z-index: 9998; position: absolute; left: 0px; top: 0px; box-shadow: 0px 0px 0px; background-color: transparent;" src="S0167865510000954_11.html" frameborder="0"></iframe></div><img src="pageReport" style="display: none;" height="0" width="0"><div id="sgfGhost" style="display: none;"><div class="containerApplOver invisible" id="sgfGhost_299152"><div id="footergadgets-gadget-title-bar-2" class="curveBoxleftNormal ActiveGadget  sgfOpen"><img src="icon.jpg" class="headingGadgetIcon " title="Entitle Gadget" border="none"><h3 style="cursor: default;" title="" id="GadgetName2" class="elsevierTimeLOver truncateTitle">Entitle Gadget</h3><div id="Controls_2" class="Controls menuControlParent" align="right"><ul class="menu"><li class="menuControls"> <a href="javascript:void(0);" onclick="shindig.container.getGadget(2).handleToggle(this);"><div id="toggleBtn_2" class="sprite-close openCloseBtn" title="Close This Application" style="background-image: url('gsframeworksprite.png');"></div></a></li><li class="menuControls"><a href="javascript:void(0);" id="button-remote_iframe_2_userPrefsDialog" onclick="shindig.container.getGadget(2).openMenuOptions();return false;"><div class="sprite-options" title="Open Options" style="background-image: url('gsframeworksprite.png');"></div></a></li><ul class="submenu" id="sb-remote_iframe_2_userPrefsDialog"></ul></ul></div><div id="Close_2" class="Controls_close" style="display: none;"><a href="javascript:shindig.container.getGadget(2).closeGadget();"><div class="sprite-close_hoover" title="Close" style="background-image: url('gsframeworksprite.png');"></div></a></div></div><div id="remote_iframe_2_userPrefsDialog" style="display: none; z-index: 10; left: 0px; top: 0px; border-width: 0px 1px 1px; border-style: none solid solid; border-color: -moz-use-text-color rgb(205, 240, 164) rgb(205, 240, 164); -moz-border-top-colors: none; -moz-border-right-colors: none; -moz-border-bottom-colors: none; -moz-border-left-colors: none; border-image: none; background-color: rgb(238, 247, 227); width: 99%; font-weight: bold;" class=""></div><div id="maincontremote_iframe_2" style="border-style: solid;" class="containerAppDetails "><iframe id="remote_iframe_2" onload="sgf.killTimer('299152', 'sgfGhost');" name="remote_iframe_2" src="S0167865510000954_12.html" style="text-align: left;" "="" scrolling="auto" frameborder="no" height="" width="100%"></iframe></div></div></div></body>
</html>
